{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8e64154f",
   "metadata": {},
   "source": [
    "### Webscraper for extracting newly published names, strains and accession numbers from the weeekly IJSEM email (saved in html). Script then compares the IJSEM names to the NCBI names and generates a report used for taxonomy updates. \n",
    "### Uses Natural Language processing combined with REGEX to extract information from text\n",
    "### Notes on selenium. There is a lot of bad advise in stack overflow that refers to old versions of selenium. Need to refer to the offocial documentation https://selenium-python.readthedocs.io/ If you run out of space in your linux directory the script will crash. There is a cache file written by selenium that may then need to be deleted to get the chrome driver working again. Follow the path in the error message and delete that cache directory. Selenium needs a webdriver installed, find details at https://sites.google.com/chromium.org/driver/downloads\n",
    "\n",
    "### NOTE selenium cache files are quite large and need to be cleaned up periodically"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4657a12a",
   "metadata": {},
   "outputs": [],
   "source": [
    "pip install \"numpy<2.0\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0568a50",
   "metadata": {},
   "outputs": [],
   "source": [
    "pip install spacy\n",
    "#pip install -U pip setuptools wheel\n",
    "#pip install -U spacy\n",
    "#pip install --upgrade spacy"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec33dae7",
   "metadata": {},
   "source": [
    "### Install the model at the command lines\n",
    "small model\n",
    "python -m spacy download en_core_web_sm\n",
    "\n",
    "medium model\n",
    "python -m spacy download en_core_web_md\n",
    "\n",
    "large model\n",
    "python -m spacy download en_core_web_lg"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18c910a4",
   "metadata": {},
   "source": [
    "### create training set https://spacy.io/usage/training\n",
    "\n",
    "This annotator works better https://agateteam.org/spacynerannotate/ combined with the DocBin code to create the .spacy file. \n",
    "\n",
    "https://spacy.io/usage/training#quickstart\n",
    "\n",
    "python -m spacy init fill-config ./base_config.cfg ./config.cfg\n",
    "\n",
    "This NER annotator worked better than the one above. Used the DocBin technique to convert to .spacy file. Spacy training command with config.cfg then worked.\n",
    "\n",
    "Config.cfg needs to be created for the spacy command line training. \n",
    "\n",
    "Follow the instructions in the quickstart for the base_config.cfg file. Selected OS and clicked NER\n",
    "\n",
    "https://spacy.io/usage/training#quickstart\n",
    "\n",
    "Run the following at command line to create the config.cfg. \n",
    "\n",
    "python -m spacy init fill-config ./base_config.cfg ./config.cfg\n",
    "\n",
    "Once the config file was done and spacy file was created run the following at command line to train the model\n",
    "\n",
    "To debug the data:\n",
    "python -m spacy debug data config.cfg --paths.train ./train.spacy --paths.dev ./train.spacy\n",
    "\n",
    "To debug the config file file:\n",
    "python -m spacy train config.cfg --output ./output\n",
    "\n",
    "or\n",
    "\n",
    "python -m spacy train config.cfg \n",
    "\n",
    "train the model at command line\n",
    "python -m spacy train config.cfg --output ./output --paths.train ./train.spacy --paths.dev ./train.spacy\n",
    "\n",
    "creates two directories model-best and model-last. I used model-best \n",
    "\n",
    "nlp1 = spacy.load(r\"./output/model-best\")\n",
    "\n",
    "\n",
    "Notes\n",
    "python -m spacy init fill-config ./base_config.cfg ./config.cfg\n",
    "\n",
    "python -m spacy train config.cfg --output ./output --paths.train ./train.spacy --paths.dev ./dev.spacy\n",
    "\n",
    "https://spacy.io/usage/training#basics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1eb1d9ef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "884b2140",
   "metadata": {},
   "source": [
    "### Train spacy if not already done. Using output https://arunmozhi.in/ner-annotator/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fb2accf",
   "metadata": {},
   "outputs": [],
   "source": [
    "from spacy.tokens import DocBin\n",
    "import pandas as pd\n",
    "import json\n",
    "import os"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7dcd3654",
   "metadata": {},
   "source": [
    "### train the model at command line\n",
    "python -m spacy train config.cfg --output ./output --paths.train ./train.spacy --paths.dev ./dev.spacy "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5dfb45c3",
   "metadata": {},
   "source": [
    "### Main webscraper program start\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a58e07e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import selenium\n",
    "from selenium import webdriver\n",
    "from selenium.webdriver.common.by import By\n",
    "#PATH = \"C:\\Users\\mcveigh\\Documents\\PythonPC\\chrome.exe\"\n",
    "#driver = webdriver.Chrome(PATH)\n",
    "from selenium.webdriver.chrome.options import Options\n",
    "import time\n",
    "import IPython\n",
    "import pandas as pd\n",
    "import re\n",
    "import os\n",
    "import sys\n",
    "import bs4\n",
    "from bs4 import BeautifulSoup\n",
    "import requests\n",
    "import numpy as np\n",
    "from datetime import datetime\n",
    "from nameparser import HumanName\n",
    "import spacy\n",
    "#nlp = spacy.load(\"en_core_web_sm\")\n",
    "nlp = spacy.load(\"en_core_web_md\")\n",
    "from spacy.matcher import PhraseMatcher\n",
    "from spacy import displacy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c8abb98",
   "metadata": {},
   "outputs": [],
   "source": [
    "#test\n",
    "text2 = (\"The type strain is 2205BS29-5T (=LMG 33062T =KACC 23240T), which is isolated from a marine sponge, P. elegans, \"\n",
    "\"collected from Beomseom in Jeju-Island, Republic of Korea.\"\n",
    "\n",
    "\"The DNA G+C content of strain 2205BS29-5T is 67.8%. The GenBank accession numbers for the 16S rRNA gene and\" \n",
    "\"whole-genome sequences of strain 2205BS29-5T are OQ569368 and JAVAMQ000000000, respectively.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fbfe8be",
   "metadata": {},
   "outputs": [],
   "source": [
    "#test but need this later\n",
    "import IPython\n",
    "nlp1 = spacy.load(r\"./output/model-best\") #load the best model\n",
    "doc = nlp1(text2)\n",
    "from IPython.display import display\n",
    "from spacy import displacy\n",
    "from IPython.core.display import HTML, display\n",
    "displacy.render(doc, style=\"ent\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a6a4c13",
   "metadata": {},
   "source": [
    "### Find URLS from saved email in html - save as from outlook in htm format. URLs are extracted and saved as input for selenium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6136fcf9",
   "metadata": {},
   "outputs": [],
   "source": [
    "input = (r'IJSEMemail54.htm')\n",
    "output = (r'NameCheckweek54.xlsx')\n",
    "alldescriptions = (r'all_descriptions54')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8a2d86b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Only need this if you suspect a problem with beautiful soup not imporing the URL list correctly. Otherwise skip this step\n",
    "from bs4.diagnose import diagnose\n",
    "with open (input, encoding = 'unicode_escape') as f:\n",
    "    data = f.read()\n",
    "diagnose(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "58762373",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "8c2ca58b-96ca-48f8-8989-7101612b38eb",
   "metadata": {},
   "source": [
    "#### alternative Beautifiul soup code for extracting URLS with autodetect encoding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8169ef07",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 13 URLs:\n",
      "http://dx.doi.org/10.1099/ijsem.0.006836\n",
      "http://dx.doi.org/10.1099/ijsem.0.006850\n",
      "http://dx.doi.org/10.1099/ijsem.0.006839\n",
      "http://dx.doi.org/10.1099/ijsem.0.006837\n",
      "http://dx.doi.org/10.1099/ijsem.0.006840\n",
      "http://dx.doi.org/10.1099/ijsem.0.006753\n",
      "http://dx.doi.org/10.1099/ijsem.0.006834\n",
      "http://dx.doi.org/10.1099/ijsem.0.006814\n",
      "http://dx.doi.org/10.1099/ijsem.0.006819\n",
      "http://dx.doi.org/10.1099/ijsem.0.006813\n",
      "http://dx.doi.org/10.1099/ijsem.0.006833\n",
      "http://dx.doi.org/10.1099/ijsem.0.006828\n",
      "http://dx.doi.org/10.1099/ijsem.0.006838\n"
     ]
    }
   ],
   "source": [
    "#alternative Beautifiul soup with autodetect encoding\n",
    "from charset_normalizer import from_path\n",
    "\n",
    "# Auto-detect file encoding\n",
    "result = from_path(input).best()\n",
    "html = str(result)\n",
    "\n",
    "# Parse with BeautifulSoup\n",
    "soup = BeautifulSoup(html, \"html.parser\")\n",
    "text = soup.get_text()\n",
    "\n",
    "# Extract all http/https links\n",
    "urls = re.findall(r\"https?://\\S+\", text)\n",
    "urls = [url.rstrip('.,);') for url in urls]\n",
    "\n",
    "# Filter links as needed\n",
    "filtered_urls = [\n",
    "    url for url in urls\n",
    "    if all(exclude not in url for exclude in [\"TandC\", \"myaccount\", \"join-the-society\"])\n",
    "    # Remove \"doi.org\" from filter if you want article links\n",
    "]\n",
    "\n",
    "if not filtered_urls:\n",
    "    raise ValueError(\"No usable URLs found!\")\n",
    "\n",
    "print(f\"Found {len(filtered_urls)} URLs:\")\n",
    "print(\"\\n\".join(filtered_urls))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7e20bf4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f40419a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_strains(description):\n",
    "    \"\"\"Find strains with spacy\"\"\"\n",
    "    #print('Entering strain subroutine')\n",
    "    doc = nlp(description)\n",
    "    nlp1 = spacy.load(r\"./output/model-best\") #load the best model\n",
    "    phrase_matcher = PhraseMatcher(nlp.vocab)\n",
    "    phrases = ['strain', 'Strain', 'strains', 'Strains']\n",
    "    patterns = [nlp(description) for description in phrases]\n",
    "    phrase_matcher.add('STRAIN', None, *patterns)\n",
    "               \n",
    "    for sent in doc.sents:    \n",
    "        for match_id, start, end in phrase_matcher(nlp(sent.text)):\n",
    "            if nlp.vocab.strings[match_id] in [\"STRAIN\"]:\n",
    "                print(sent.text)\n",
    "                doc = nlp1(sent.text)\n",
    "                displacy.render(doc, style=\"ent\")\n",
    "                for entity in doc.ents:\n",
    "                    if entity.text not in strains:\n",
    "                        entity.text.encode('ascii', 'ignore').decode('unicode_escape')\n",
    "                        strains.append(entity.text)\n",
    "    return strains\n",
    "\n",
    "def remove_non_ascii(text):\n",
    "    \"\"\"Remove non-ASCII characters\"\"\"\n",
    "    return ''.join(char for char in text if ord(char) < 128)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33443ded",
   "metadata": {},
   "source": [
    "### Main body - Selenium to extract data from html\n",
    "### https://nameparser.readthedocs.io/en/latest/index.html to find the last name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "22d6f4bf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rhodotorula silvicola sp. nov., a new yeast species from plant-associated substrates and mushroom\n",
      "orgname ['F.Y. Bai ', 'R. silvicola ']\n",
      "Neither pseudohyphae nor true hyphae are formed on CMA and PDA after 10days at 20C. Additionally, sexual structure was not observed in the single and pairwise mixed strain cultures of strain CGMCC 2.7770 and strain UCDFST 81-84 on different media, including CMA, PDA, YM, YCBAS and V8, after incubation at 20C for 2months.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">Neither pseudohyphae nor true hyphae are formed on CMA and PDA after 10days at 20C. Additionally, sexual structure was not observed in the single and pairwise mixed strain cultures of strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CGMCC 2.7770\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    UCDFST 81-84\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " on different media, including CMA, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    PDA\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    YM\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ", YCBAS and V8, after incubation at 20C for 2months.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Neither pseudohyphae nor true hyphae are formed on CMA and PDA after 10days at 20C. Additionally, sexual structure was not observed in the single and pairwise mixed strain cultures of strain CGMCC 2.7770 and strain UCDFST 81-84 on different media, including CMA, PDA, YM, YCBAS and V8, after incubation at 20C for 2months.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">Neither pseudohyphae nor true hyphae are formed on CMA and PDA after 10days at 20C. Additionally, sexual structure was not observed in the single and pairwise mixed strain cultures of strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CGMCC 2.7770\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    UCDFST 81-84\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " on different media, including CMA, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    PDA\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    YM\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ", YCBAS and V8, after incubation at 20C for 2months.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Neither pseudohyphae nor true hyphae are formed on CMA and PDA after 10days at 20C. Additionally, sexual structure was not observed in the single and pairwise mixed strain cultures of strain CGMCC 2.7770 and strain UCDFST 81-84 on different media, including CMA, PDA, YM, YCBAS and V8, after incubation at 20C for 2months.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">Neither pseudohyphae nor true hyphae are formed on CMA and PDA after 10days at 20C. Additionally, sexual structure was not observed in the single and pairwise mixed strain cultures of strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CGMCC 2.7770\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    UCDFST 81-84\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " on different media, including CMA, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    PDA\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    YM\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ", YCBAS and V8, after incubation at 20C for 2months.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The morphology of R. silvicola sp. nov. strain CGMCC 2.7770T observed on YPD agar after 1week at 20C includes colony morphology (a) and vegetative cells (b).\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The morphology of R. silvicola sp. nov. strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CGMCC 2.7770T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " observed on YPD agar after 1week at 20C includes colony morphology (a) and vegetative cells (b).</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Growth on YPD agar is positive at 30C but negative at 37C.\n",
      "The holotype, CGMCC 2.7770T (original strain number 1017-17-3), was isolated from rotten leaves from Zunyi City, Guizhou Province, PR China, in September 2023, and has been deposited in a metabolically inactive state in the China General Microbiological Culture Collection Center, Beijing, PR China.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">Growth on YPD agar is positive at 30C but negative at 37C.<br>The holotype, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CGMCC 2.7770T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (original strain number \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    1017-17-3\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from rotten leaves from Zunyi City, Guizhou Province, PR China, in September 2023, and has been deposited in a metabolically inactive state in the China General Microbiological Culture Collection Center, Beijing, PR China.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The GenBank accession number for the 26S rDNA D1/D2 domains and the ITS region sequences of the type strain 1017-17-3 is PQ432548 and PQ432547, respectively.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The GenBank accession number for the 26S rDNA D1/D2 domains and the ITS region sequences of the type strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    1017-17-3\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " is PQ432548 and PQ432547, respectively.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['CGMCC 2.7770', 'UCDFST 81-84', 'PDA', 'YM', 'CGMCC 2.7770T', '1017-17-3']\n",
      "orgname ['F.Y. Bai ']\n",
      "strain names ['CGMCC 2.7770', 'UCDFST 81-84', 'PDA', 'YM', 'CGMCC 2.7770T', '1017-17-3']\n",
      "BREAK1\n",
      "Farysia photiniae sp. nov., an anamorphic yeast species isolated from plant leaves\n",
      "Outer div ID: s3-4, Text: Description of Farysia photiniae sp. nov. C.-Y. Chai & F.-L. Hui\n",
      "N.L. gen. n. photiniae, referring to Photinia, the plant genus from which the type strain was isolated).\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">N.L. gen. n. photiniae, referring to Photinia, the plant genus from which the type strain was isolated).<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The ex-type culture has been deposited in a metabolically inactive state in the Portuguese Yeast Culture Collection (PYCC), Caparica, Portugal, as strain PYCC 9932 (=NYNU 22992).\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The ex-type culture has been deposited in a metabolically inactive state in the Portuguese Yeast Culture Collection (PYCC), Caparica, Portugal, as strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    PYCC 9932\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    NYNU 22992\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ").</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BREAK2\n",
      "Methanosuratincola petrocarbonis gen. nov., sp. nov., a methyl-reducing methanogen isolated from Shengli oil field, and proposal of Methanosuratincolaceae fam. nov., Methanosuratincolales ord. nov. and Methanosuratincolia classis nov. in the phylum Thermoproteota\n",
      "Outer div ID: s4-1, Text: Description of Methanosuratincola gen. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-2, Text: Description of Methanosuratincola petrocarbonis sp. nov.\n",
      "The type strain, LWZ-6T (=CCAM 1872T=JCM 39528T), was isolated from oily sludge of the Shengli oil field in China.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    LWZ-6T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CCAM 1872T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 39528T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from oily sludge of the Shengli oil field in China.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BREAK2\n",
      "Outer div ID: s4-3, Text: Description of Methanosuratincolaceae fam. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-4, Text: Description of Methanosuratincolales ord. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-5, Text: Description of Methanosuratincolia classis nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-1, Text: Description of Methanosuratincola gen. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-2, Text: Description of Methanosuratincola petrocarbonis sp. nov.\n",
      "The type strain, LWZ-6T (=CCAM 1872T=JCM 39528T), was isolated from oily sludge of the Shengli oil field in China.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    LWZ-6T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CCAM 1872T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 39528T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from oily sludge of the Shengli oil field in China.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BREAK2\n",
      "Outer div ID: s4-3, Text: Description of Methanosuratincolaceae fam. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-4, Text: Description of Methanosuratincolales ord. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-5, Text: Description of Methanosuratincolia classis nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-1, Text: Description of Methanosuratincola gen. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-2, Text: Description of Methanosuratincola petrocarbonis sp. nov.\n",
      "The type strain, LWZ-6T (=CCAM 1872T=JCM 39528T), was isolated from oily sludge of the Shengli oil field in China.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    LWZ-6T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CCAM 1872T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 39528T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from oily sludge of the Shengli oil field in China.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BREAK2\n",
      "Outer div ID: s4-3, Text: Description of Methanosuratincolaceae fam. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-4, Text: Description of Methanosuratincolales ord. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-5, Text: Description of Methanosuratincolia classis nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-1, Text: Description of Methanosuratincola gen. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-2, Text: Description of Methanosuratincola petrocarbonis sp. nov.\n",
      "The type strain, LWZ-6T (=CCAM 1872T=JCM 39528T), was isolated from oily sludge of the Shengli oil field in China.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    LWZ-6T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CCAM 1872T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 39528T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from oily sludge of the Shengli oil field in China.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BREAK2\n",
      "Outer div ID: s4-3, Text: Description of Methanosuratincolaceae fam. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-4, Text: Description of Methanosuratincolales ord. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-5, Text: Description of Methanosuratincolia classis nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-1, Text: Description of Methanosuratincola gen. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-2, Text: Description of Methanosuratincola petrocarbonis sp. nov.\n",
      "The type strain, LWZ-6T (=CCAM 1872T=JCM 39528T), was isolated from oily sludge of the Shengli oil field in China.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    LWZ-6T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CCAM 1872T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 39528T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from oily sludge of the Shengli oil field in China.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BREAK2\n",
      "Outer div ID: s4-3, Text: Description of Methanosuratincolaceae fam. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-4, Text: Description of Methanosuratincolales ord. nov.\n",
      "BREAK2\n",
      "Outer div ID: s4-5, Text: Description of Methanosuratincolia classis nov.\n",
      "BREAK2\n",
      "Brachymonas wangyanguii sp. nov., isolated from the coal samples in Shanxi Province of China\n",
      "orgname ['Brachymonas wangyanguii ']\n",
      "The type strain G13T (GDMCC 1.4691T=JCM 37746T) was obtained from an in situ coal block sample obtained from Xinzhou, Shanxi, China.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    G13T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    GDMCC 1.4691T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 37746T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ") was obtained from an in situ coal block sample obtained from Xinzhou, Shanxi, China.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The genomic DNA of this strain exhibited a G+C content of 63.5 mol%.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The genomic DNA of this strain exhibited a G+C content of 63.5 mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['G13T', 'GDMCC 1.4691T', 'JCM 37746T']\n",
      "orgname ['Brachymonas wangyanguii ']\n",
      "strain names ['G13T', 'GDMCC 1.4691T', 'JCM 37746T']\n",
      "BREAK1\n",
      "Fodinibius alkaliphilus sp. nov., a moderately halophilic and alkaliphilic bacterium isolated from an inland saltern in central Portugal and reclassification of Aliifodinibius salipaludis as Fodinibius salipaludis sp. nov.\n",
      "orgname ['Fodinibius alkaliphilus ']\n",
      "The type strain N2T (KCTC 102228T=MCCC 1K08942T) was isolated from a water sample collected from an inland saltern in Rio Maior, central Portugal.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    N2T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    KCTC 102228T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    MCCC 1K08942T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ") was isolated from a water sample collected from an inland saltern in Rio Maior, central Portugal.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The genome of the type strain has an approximate size of 3.41Mb and a G+C content of 42.0mol%.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The genome of the type strain has an approximate size of 3.41Mb and a G+C content of 42.0mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['N2T', 'KCTC 102228T', 'MCCC 1K08942T']\n",
      "orgname ['Fodinibius alkaliphilus ']\n",
      "strain names ['N2T', 'KCTC 102228T', 'MCCC 1K08942T']\n",
      "BREAK1\n",
      "orgname ['Fodinibius salipaludis ']\n",
      "The type strain is WN023T (=KCTC 52855T=ACCC 19978T).\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain is \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    WN023T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    KCTC 52855T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ACCC 19978T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ").</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The genome of the type strain has an approximate size of 3.58Mb and has a G+C content of 42.2mol%.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The genome of the type strain has an approximate size of 3.58Mb and has a G+C content of 42.2mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['WN023T', 'KCTC 52855T', 'ACCC 19978T']\n",
      "orgname ['Fodinibius salipaludis ']\n",
      "strain names ['WN023T', 'KCTC 52855T', 'ACCC 19978T']\n",
      "BREAK1\n",
      "Chlamydia vaughanii sp. nov., a novel Chlamydia isolated from a tropical fish (bushymouth catfish)\n",
      "orgname ['Chlamydia vaughanii ']\n",
      "The type strain is an intracellular bacterium that could be grown in mouse (McCoy, ATCC CRL-1696) and fathead minnow (EPC, ATCC CRL-2872) cells incubated at 37 and 30C, respectively.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain is an intracellular bacterium that could be grown in mouse (McCoy, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ATCC CRL-1696\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ") and fathead minnow (\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    EPC\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ", \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ATCC CRL-2872\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ") cells incubated at 37 and 30C, respectively.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The genome of strain CRIB76 has a size of 1.3 Mbp and a 7.6 Kbp plasmid, with a G+C content of 39.3mol%.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The genome of strain CRIB76 has a size of 1.3 Mbp and a 7.6 Kbp plasmid, with a G+C content of 39.3mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C. vaughanii strain CRIB76\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">C. vaughanii strain CRIB76</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "QA1836) strain collections.\n",
      " \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">QA1836) strain collections.<br> </div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['ATCC CRL-1696', 'EPC', 'ATCC CRL-2872']\n",
      "orgname ['Chlamydia vaughanii ']\n",
      "strain names ['ATCC CRL-1696', 'EPC', 'ATCC CRL-2872']\n",
      "BREAK1\n",
      "Aestuariibius violaceus sp. nov., isolated from a marine limpet Cellana toreuma\n",
      "orgname ['Aestuariibius violaceus ']\n",
      "The type strain, 2305UL40-4T (=KCCM 43505T=MCCC 1K09164T), was isolated from C. toreuma collected at Ulleungdo, Republic of Korea.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    2305UL40-4T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    KCCM 43505T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    MCCC 1K09164T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from C. toreuma collected at Ulleungdo, Republic of Korea.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['2305UL40-4T', 'KCCM 43505T', 'MCCC 1K09164T']\n",
      "orgname ['Aestuariibius violaceus ']\n",
      "strain names ['2305UL40-4T', 'KCCM 43505T', 'MCCC 1K09164T']\n",
      "BREAK1\n",
      "Characterization of four novel members of genus Streptococcus from wild animals: description of Streptococcus handemini sp. nov., Streptococcus jiangjianxini sp. nov., Streptococcus tangpeifui sp. nov. and Streptococcus huangxiaojuni sp. nov.\n",
      "orgname ['Streptococcus handemini ']\n",
      "The type strain, ZJ93T (=GDMCC 1.3112T=JCM 35325T), was isolated from the respiratory tracts of marmots.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ZJ93T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    GDMCC 1.3112T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 35325T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from the respiratory tracts of marmots.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The DNA G+C content of this strain is 41.0mol%.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The DNA G+C content of this strain is 41.0mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ZJ100 is a second strain of the species.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">ZJ100 is a second strain of the species.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The GenBank/EMBL/DDBJ accession numbers for the 16S rRNA gene sequences of strains ZJ93 and ZJ100 are OL504969 and PP830069, respectively.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The GenBank/EMBL/DDBJ accession numbers for the 16S rRNA gene sequences of strains \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ZJ93\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and ZJ100 are OL504969 and PP830069, respectively.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The genome sequences of these strains are available under accession numbers JBFRYH000000000 and JBEHGM000000000.\n",
      " \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The genome sequences of these strains are available under accession numbers JBFRYH000000000 and JBEHGM000000000.<br> </div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['ZJ93T', 'GDMCC 1.3112T', 'JCM 35325T', 'ZJ93']\n",
      "orgname ['Streptococcus handemini ']\n",
      "strain names ['ZJ93T', 'GDMCC 1.3112T', 'JCM 35325T', 'ZJ93']\n",
      "BREAK1\n",
      "orgname ['Streptococcus jiangjianxini ']\n",
      "The type strain, ZJ151T (=GDMCC 1.3116T=JCM 35326T), was isolated from the respiratory tracts of marmots.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ZJ151T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    GDMCC 1.3116T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 35326T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from the respiratory tracts of marmots.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The DNA G+C content of this strain is 39.0mol%.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The DNA G+C content of this strain is 39.0mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ZJ99 is a second strain of the species.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ZJ99\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " is a second strain of the species.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The GenBank/EMBL/DDBJ accession numbers for the 16S rRNA gene sequences of strains ZJ151T and ZJ99 are OL504968 and PP830068, respectively.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The GenBank/EMBL/DDBJ accession numbers for the 16S rRNA gene sequences of strains \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ZJ151T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ZJ99\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " are OL504968 and PP830068, respectively.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The genome sequences of these strains are available under accession numbers JBEHGJ000000000 and JBEHGL000000000.\n",
      " \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The genome sequences of these strains are available under accession numbers JBEHGJ000000000 and JBEHGL000000000.<br> </div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['ZJ151T', 'GDMCC 1.3116T', 'JCM 35326T', 'ZJ99']\n",
      "orgname ['Streptococcus jiangjianxini ']\n",
      "strain names ['ZJ151T', 'GDMCC 1.3116T', 'JCM 35326T', 'ZJ99']\n",
      "BREAK1\n",
      "orgname ['Streptococcus tangpeifui ']\n",
      "The type strain, ZJ1593T (=GDMCC 1.3115T=JCM 35327T), was isolated from the respiratory tracts of marmots.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ZJ1593T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    GDMCC 1.3115T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 35327T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from the respiratory tracts of marmots.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The DNA G+C content of this strain is 41.5mol%.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The DNA G+C content of this strain is 41.5mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ZJ373 is a second strain of the species.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">ZJ373 is a second strain of the species.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The GenBank/EMBL/DDBJ accession numbers for the 16S rRNA gene sequences of strains ZJ1593T and ZJ373 are OL504562 and MN850392, respectively.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The GenBank/EMBL/DDBJ accession numbers for the 16S rRNA gene sequences of strains \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ZJ1593T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ZJ373\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " are OL504562 and MN850392, respectively.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The genome sequences of these strains are available under accession numbers CP172427 and JAAKDU000000000.\n",
      " \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The genome sequences of these strains are available under accession numbers CP172427 and JAAKDU000000000.<br> </div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['ZJ1593T', 'GDMCC 1.3115T', 'JCM 35327T', 'ZJ373']\n",
      "orgname ['Streptococcus tangpeifui ']\n",
      "strain names ['ZJ1593T', 'GDMCC 1.3115T', 'JCM 35327T', 'ZJ373']\n",
      "BREAK1\n",
      "orgname ['Streptococcus huangxiaojuni ']\n",
      "The type strain, H31T (=GDMCC 1.4758T=JCM 37397T), was isolated from the faecal samples of white-lipped deer.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain, \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    H31T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    GDMCC 1.4758T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "=\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    JCM 37397T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), was isolated from the faecal samples of white-lipped deer.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The DNA G+C content of this strain is 41.5mol%.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The DNA G+C content of this strain is 41.5mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H49 is a second strain of the species.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">\n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    H49\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " is a second strain of the species.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The GenBank/EMBL/DDBJ accession numbers for the 16S rRNA gene sequences of strains H31T and H49 are PQ001808 and PQ198380, respectively.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The GenBank/EMBL/DDBJ accession numbers for the 16S rRNA gene sequences of strains \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    H31T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    H49\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " are PQ001808 and PQ198380, respectively.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The genome sequences of these strains are available under accession numbers JBFQEG000000000 and JBFQEH000000000.\n",
      " \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The genome sequences of these strains are available under accession numbers JBFQEG000000000 and JBFQEH000000000.<br> </div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['H31T', 'GDMCC 1.4758T', 'JCM 37397T', 'H49']\n",
      "orgname ['Streptococcus huangxiaojuni ']\n",
      "strain names ['H31T', 'GDMCC 1.4758T', 'JCM 37397T', 'H49']\n",
      "BREAK1\n",
      "Judicial Opinion 132\n",
      "Legionella sheltonii sp. nov., a novel species isolated on a cruise ship during routine monitoring\n",
      "orgname ['Legionella sheltonii ', 'L. sheltonii ']\n",
      "Gene sequences of the newly proposed species showed the highest similarity with those of L. cherrii (NCTC strain 11976T), including 98.7% for 16S rRNA gene, 98.4% for mip, 94.2% for gyrA, 96.2% for rpoB and 99.2% for rnpB. The POCP value for the L. sheltonii sp. nov.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">Gene sequences of the newly proposed species showed the highest similarity with those of L. cherrii (NCTC strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    11976T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       "), including 98.7% for 16S rRNA gene, 98.4% for mip, 94.2% for gyrA, 96.2% for rpoB and 99.2% for rnpB. The POCP value for the L. sheltonii sp. nov.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The whole-genome shotgun sequencing project of the strain PATHC038T has been deposited at NCBI under the genome accession number GCF_026191355.1 with a genome size of 4.3Mb and a G+C content of 38.2mol%.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The whole-genome shotgun sequencing project of the strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    PATHC038T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " has been deposited at NCBI under the genome accession number GCF_026191355.1 with a genome size of 4.3Mb and a G+C content of 38.2mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The GenBank accession numbers for the 16S rRNA genes of the strains PATHC038T and HB10 are PQ120583 and PQ120582, respectively.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The GenBank accession numbers for the 16S rRNA genes of the strains \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    PATHC038T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    HB10\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " are PQ120583 and PQ120582, respectively.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The GenBank accession numbers for the mip genes of the strains PATHC038T and HB10 are PQ119488 and PQ119489, respectively.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The GenBank accession numbers for the mip genes of the strains \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    PATHC038T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    HB10\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " are PQ119488 and PQ119489, respectively.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Type strain PATHC038T culture collection deposit numbers are CCUG 76918T and ATCC TSD-370.\n",
      " \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">Type strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    PATHC038T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " culture collection deposit numbers are \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    CCUG 76918T\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " and \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    ATCC TSD-370\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       ".<br> </div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['11976T', 'PATHC038T', 'HB10', 'CCUG 76918T', 'ATCC TSD-370']\n",
      "orgname ['Legionella sheltonii ']\n",
      "strain names ['11976T', 'PATHC038T', 'HB10', 'CCUG 76918T', 'ATCC TSD-370']\n",
      "BREAK1\n",
      "Emendation to Rule 8 of the International Code of Nomenclature of Prokaryotes to clarify the meaning of the term ‘stem’\n",
      "Oceaniferula spumae sp. nov., a novel Verrucomicrobiota bacterium, isolated from sea foam at Noto Peninsula, Ishikawa, Japan\n",
      "orgname ['Oceaniferula spumae ']\n",
      "L. gen. n. spumae, of foam, denoting the isolation of the strain from sea foam on the Maura Coast of the Noto Peninsula, Japan) is a bacterium isolated from sea foam.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mcveigh/jupweb/lib/python3.11/site-packages/spacy/displacy/__init__.py:213: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.\n",
      "  warnings.warn(Warnings.W006)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">L. gen. n. spumae, of foam, denoting the isolation of the strain from sea foam on the Maura Coast of the Noto Peninsula, Japan) is a bacterium isolated from sea foam.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cells of strain NT6NT are Gram-stain-negative, aerobic, rod-shaped and ~0.50.6m in width and 1.06.0m in length.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">Cells of strain NT6NT are Gram-stain-negative, aerobic, rod-shaped and ~0.50.6m in width and 1.06.0m in length.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The growth conditions for strain NT6NT are 1530C (optimum, 2530C), 3.5% NaCl (w/v) (1.55.0%) and pH 7.09.0 (pH 8.0).\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The growth conditions for strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    NT6NT\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " are 1530C (optimum, 2530C), 3.5% NaCl (w/v) (1.55.0%) and pH 7.09.0 (pH 8.0).</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The major polar lipids of strain NT6NT were PE, DPG and PG.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The major polar lipids of strain NT6NT were PE, DPG and PG.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The G+C content of the genomic DNA of strain NT6NT is 52.1mol%.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The G+C content of the genomic DNA of strain NT6NT is 52.1mol%.</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The DDBJ/GenBank/EMBL accession numbers for the 16S rRNA gene sequence and whole-genome sequence of strain NT6NT are LC820896 and AP026866.1, respectively.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The DDBJ/GenBank/EMBL accession numbers for the 16S rRNA gene sequence and whole-genome sequence of strain \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    NT6NT\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " are LC820896 and AP026866.1, respectively.<br></div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The type strain is NT6NT (=\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<span class=\"tex2jax_ignore\"><div class=\"entities\" style=\"line-height: 2.5; direction: ltr\">The type strain is \n",
       "<mark class=\"entity\" style=\"background: #ddd; padding: 0.45em 0.6em; margin: 0 0.25em; line-height: 1; border-radius: 0.35em;\">\n",
       "    NT6NT\n",
       "    <span style=\"font-size: 0.8em; font-weight: bold; line-height: 1; border-radius: 0.35em; vertical-align: middle; margin-left: 0.5rem\">STRAIN</span>\n",
       "</mark>\n",
       " (=</div></span>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "strain names ['NT6NT']\n",
      "orgname ['Oceaniferula spumae ']\n",
      "strain names ['NT6NT']\n",
      "BREAK1\n",
      "Corrigendum: Faecalibacterium langellae sp. nov. isolated from human faeces\n"
     ]
    }
   ],
   "source": [
    "#from selenium.webdriver.common.by import By\n",
    "pub_df = pd.DataFrame(columns=['PublishedName', 'Accessions', 'Strains', 'Authority', 'DOI', 'filtered_url'])\n",
    "pd.set_option('display.max_columns', None)\n",
    "combined_description = []\n",
    "for filtered_url in filtered_urls:\n",
    "    counter = 1\n",
    "    strains = []\n",
    "    accessions = []\n",
    "    orgname = []\n",
    "    doi = []\n",
    "    author = []\n",
    "    date = []\n",
    "    year = []\n",
    "    author_raw = []\n",
    "    author = []\n",
    "    name1 = []\n",
    "    name2 = []\n",
    "    author_count = []\n",
    "    authority = []\n",
    "    description = None\n",
    "    description1 = []\n",
    "    description2 = []\n",
    "    snumber = []\n",
    "    #scrape_with_selenium(filtered_url, counter)\n",
    "    #scrape_with_beautifulsoup(filtered_url)\n",
    "    options = Options()\n",
    "    options.headless = False  # Set to True for headless mode\n",
    "    driver = webdriver.Chrome(options=options)\n",
    "\n",
    "    #Navigate to the webpage\n",
    "    driver.get(filtered_url)\n",
    "\n",
    "    #Allow time for dynamic content to load (you may need to use WebDriverWait for more robust waiting)\n",
    "    time.sleep(3)\n",
    "    \n",
    "    html = driver.page_source\n",
    "    soup = BeautifulSoup(html)\n",
    "    \n",
    "    for element in driver.find_elements(By.CLASS_NAME, \"item-meta-data__item-title\"):\n",
    "        #print(element.text)\n",
    "        title = element.text\n",
    "        print(title)\n",
    "        \n",
    "    for element in driver.find_elements(By.PARTIAL_LINK_TEXT, \"doi.org\"):\n",
    "        doi = element.text\n",
    "        #print(doi)\n",
    "    \n",
    "    #find publication year\n",
    "    for element in driver.find_elements(By.XPATH, \"//*[@id='bellowheadercontainer']/main/div[2]/div/ul/li[3]/span/span[2]\"):\n",
    "        date = element.text\n",
    "        year = date[-4:]\n",
    "        #print(year)\n",
    "    \n",
    "    #find authors\n",
    "    for element in driver.find_elements(By.XPATH, \"//*[@id='bellowheadercontainer']/main/div[2]/div/ul/li[1]/span\"):\n",
    "        author_raw = element.text\n",
    "        #print(author_raw)\n",
    "        author = re.sub(r\"[\\d+]+\",'',author_raw)\n",
    "        author = re.sub(r\"†\",'',author)\n",
    "        author = re.sub(r\" and \",',',author)\n",
    "        author = re.sub(r\",,\",',',author)  \n",
    "        author = author.split(',')\n",
    "        author[:] = [item for item in author if item != '']\n",
    "        #print(author)\n",
    "        author_count = len(author)\n",
    "        #print(author_count)\n",
    "        if author_count == 1:\n",
    "            name1 = (str(author[0])) \n",
    "            name1 = HumanName(name1)\n",
    "            authority = name1.last + ' ' +str(year)\n",
    "            #print(authority)\n",
    "        elif author_count ==2:\n",
    "            name1 = (str(author[0]))\n",
    "            name1 = HumanName(name1)\n",
    "            name2 = (str(author[1]))\n",
    "            name2 = HumanName(name2)\n",
    "            authority = name1.last + ' and ' + name2.last + ' ' +str(year)\n",
    "            #print(authority)\n",
    "        else:\n",
    "            name1 = (str(author[0]))\n",
    "            name1 = HumanName(name1)\n",
    "            authority = name1.last + ' et al. ' +str(year)\n",
    "            #print(authority)\n",
    "    \n",
    "    #extract data from each species description\n",
    "    for element in driver.find_elements(By.CSS_SELECTOR, \"div.tl-main-part.title\"): #finds section headers\n",
    "        #print(element.text)\n",
    "        counter += 1\n",
    "        description = element.text\n",
    "        #print(description)\n",
    "        if \"Description of\" in description:\n",
    "            strains = []\n",
    "            #print('found', description)  \n",
    "            #snumber = 's' + str(counter - 4) + '/p[3]'\n",
    "            snumber = 's' + str(counter - 4)\n",
    "            #print('snumber is', snumber)\n",
    "            for element in driver.find_elements(By.ID, snumber):\n",
    "                description = element.text\n",
    "                cleaned_text = remove_non_ascii(description)\n",
    "                combined_description.append(cleaned_text)\n",
    "                #print(cleaned_text)\n",
    "                #print(description)\n",
    "                        \n",
    "                #find the organism names \n",
    "                orgname = []\n",
    "                match = [r'(\\S+\\s+){2}(?=sp. nov)', r'(\\S+\\s+){2}(?=nom. nov.)', r'(\\S+\\s+){2}(?=SP. NOV.)', r'(\\S+\\s+){4}(?=subsp. nov.)']\n",
    "                regex = re.compile(r'\\b(' + '|'.join(match) + r')\\b')\n",
    "                if description is not None:\n",
    "                    orgname = [m.group() for m in regex.finditer(description)]\n",
    "                    print('orgname', orgname)\n",
    "\n",
    "                #find the accessions\n",
    "                pattern = [r'[A-Z]{2}\\d{6}', r'[A-Z]{4}\\d{8}', r'([A-Z]+)(_[A-Z]+)\\d{6}', r'[A-Z]{6}\\d{9}']\n",
    "                regex = re.compile(r'\\b(' + '|'.join(pattern) + r')\\b')\n",
    "                if description is not None:\n",
    "                    accessions = [m.group() for m in regex.finditer(description)]\n",
    "                    #print('accessions', accessions)\n",
    "    \n",
    "                #find the strains\n",
    "                if description is not None:\n",
    "                    find_strains(cleaned_text)\n",
    "                    print('strain names', strains)\n",
    "    \n",
    "                #load data into pandas dataframe\n",
    "                row_data = [orgname, accessions, strains, authority, doi, filtered_url]\n",
    "                length = len(pub_df)\n",
    "                pub_df.loc[length] = row_data\n",
    "            print('BREAK1')\n",
    "\n",
    "   \n",
    "    for element in driver.find_elements(By.CLASS_NAME, \"tl-lowest-section\"): #finds section headers\n",
    "        description1 = element.text\n",
    "        outer_html = element.get_attribute(\"outerHTML\")\n",
    "        if \"Description of\" in description1:          \n",
    "            #print(outer_html)\n",
    "            spans = soup.find_all('span', attrs = {'class' : 'tl-lowest-section'})\n",
    "            for span in spans:\n",
    "                if \"Description of\" in span.text:   \n",
    "                    #print (span.text)\n",
    "                    outer_div_id = span.find_parent('div').get('id')\n",
    "                    print(f\"Outer div ID: {outer_div_id}, Text: {span.text}\")\n",
    "                    for element in driver.find_elements(By.ID, outer_div_id):\n",
    "                        description = element.text\n",
    "                        cleaned_text = remove_non_ascii(description)\n",
    "                        combined_description.append(cleaned_text)\n",
    "                        #print(cleaned_text)\n",
    "                        #print(description)\n",
    "                        \n",
    "                    #find the organism names   \n",
    "                    orgname = []\n",
    "                    match = [r'(\\S+\\s+){2}(?=sp. nov)', r'(\\S+\\s+){2}(?=nom. nov.)', r'(\\S+\\s+){2}(?=SP. NOV.)', r'(\\S+\\s+){4}(?=subsp. nov.)']\n",
    "                    regex = re.compile(r'\\b(' + '|'.join(match) + r')\\b')\n",
    "                    if description is not None:\n",
    "                        orgname = [m.group() for m in regex.finditer(description)]\n",
    "                        #print('orgname', orgname)\n",
    "\n",
    "                    #find the accessions\n",
    "                    pattern = [r'[A-Z]{2}\\d{6}', r'[A-Z]{4}\\d{8}', r'([A-Z]+)(_[A-Z]+)\\d{6}', r'[A-Z]{6}\\d{9}']\n",
    "                    regex = re.compile(r'\\b(' + '|'.join(pattern) + r')\\b')\n",
    "                    if description is not None:\n",
    "                        accessions = [m.group() for m in regex.finditer(description)]\n",
    "                        #print('accessions', accessions)\n",
    "    \n",
    "                    #find the strains\n",
    "                    strains = []\n",
    "                    if description is not None:\n",
    "                        find_strains(cleaned_text)\n",
    "                        #print('strain names', strains)\n",
    "    \n",
    "                    #load data into pandas dataframe\n",
    "                    row_data = [orgname, accessions, strains, authority, doi, filtered_url]\n",
    "                    length = len(pub_df)\n",
    "                    pub_df.loc[length] = row_data\n",
    "                    print('BREAK2')\n",
    "\n",
    "\n",
    "\n",
    "        \n",
    "#Close the browser window\n",
    "    driver.quit()    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8a2b16f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#optional write description to a file\n",
    "#print(combined_description)\n",
    "file = open(alldescriptions, \"w\")\n",
    "file.writelines(combined_description)\n",
    "file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "42783410",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PublishedName</th>\n",
       "      <th>Accessions</th>\n",
       "      <th>Strains</th>\n",
       "      <th>Authority</th>\n",
       "      <th>DOI</th>\n",
       "      <th>filtered_url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F.Y. Bai</td>\n",
       "      <td>[PQ432548, PQ432547]</td>\n",
       "      <td>CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3</td>\n",
       "      <td>Wei et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006836</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R. silvicola</td>\n",
       "      <td>[PQ432548, PQ432547]</td>\n",
       "      <td>CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3</td>\n",
       "      <td>Wei et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006836</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>F.Y. Bai</td>\n",
       "      <td>[]</td>\n",
       "      <td>CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3</td>\n",
       "      <td>Wei et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006836</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>[OP566885, OP566881, PV170748, PV170750, PV170746, PV172328, PV172327, PV170749, PV170751, PV170747]</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>Wu et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006839</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Methanosuratincola petrocarbonis</td>\n",
       "      <td>[OR243905]</td>\n",
       "      <td>LWZ-6T,  CCAM 1872T,  JCM 39528T</td>\n",
       "      <td>Wu et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006839</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Brachymonas wangyanguii</td>\n",
       "      <td>[PP212021, PQ632502, CP173677, JBBEWD000000000]</td>\n",
       "      <td>G13T,  GDMCC 1.4691T,  JCM 37746T</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006837</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Fodinibius alkaliphilus</td>\n",
       "      <td>[PP669793, JBCFYD000000000]</td>\n",
       "      <td>N2T,  KCTC 102228T,  MCCC 1K08942T</td>\n",
       "      <td>He et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Fodinibius salipaludis</td>\n",
       "      <td>[MF782427, NSKE00000000]</td>\n",
       "      <td>WN023T,  KCTC 52855T,  ACCC 19978T</td>\n",
       "      <td>He et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Chlamydia vaughanii</td>\n",
       "      <td>[OZ026853]</td>\n",
       "      <td>ATCC CRL-1696,  EPC,  ATCC CRL-2872</td>\n",
       "      <td>Marquis et al. 2025</td>\n",
       "      <td>https://doi.org/10.6084/m9.figshare.28424588.v</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Aestuariibius violaceus</td>\n",
       "      <td>[OR945532, JBFONT000000000]</td>\n",
       "      <td>2305UL40-4T,  KCCM 43505T,  MCCC 1K09164T</td>\n",
       "      <td>Lee et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006834</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>[OL504969, PP830069, JBFRYH000000000, JBEHGM000000000]</td>\n",
       "      <td>ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>[OL504968, PP830068, JBEHGJ000000000, JBEHGL000000000]</td>\n",
       "      <td>ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>[OL504562, MN850392, CP172427, JAAKDU000000000]</td>\n",
       "      <td>ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>[PQ001808, PQ198380, JBFQEG000000000, JBFQEH000000000]</td>\n",
       "      <td>H31T,  GDMCC 1.4758T,  JCM 37397T,  H49</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>[PQ120583, PQ120582, PQ119488, PQ119489]</td>\n",
       "      <td>11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td>Klobucar et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>L. sheltonii</td>\n",
       "      <td>[PQ120583, PQ120582, PQ119488, PQ119489]</td>\n",
       "      <td>11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td>Klobucar et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>[]</td>\n",
       "      <td>11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td>Klobucar et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Oceaniferula spumae</td>\n",
       "      <td>[LC820896, AP026866]</td>\n",
       "      <td>NT6NT</td>\n",
       "      <td>Huang et al. 2025</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006828</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006828</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        PublishedName  \\\n",
       "0                           F.Y. Bai    \n",
       "1                       R. silvicola    \n",
       "2                           F.Y. Bai    \n",
       "3                  Farysia photiniae    \n",
       "4                                 NaN   \n",
       "5   Methanosuratincola petrocarbonis    \n",
       "6            Brachymonas wangyanguii    \n",
       "7            Fodinibius alkaliphilus    \n",
       "8             Fodinibius salipaludis    \n",
       "9                Chlamydia vaughanii    \n",
       "10           Aestuariibius violaceus    \n",
       "11           Streptococcus handemini    \n",
       "12       Streptococcus jiangjianxini    \n",
       "13          Streptococcus tangpeifui    \n",
       "14       Streptococcus huangxiaojuni    \n",
       "15              Legionella sheltonii    \n",
       "16                      L. sheltonii    \n",
       "17              Legionella sheltonii    \n",
       "18               Oceaniferula spumae    \n",
       "\n",
       "                                                                                              Accessions  \\\n",
       "0                                                                                   [PQ432548, PQ432547]   \n",
       "1                                                                                   [PQ432548, PQ432547]   \n",
       "2                                                                                                     []   \n",
       "3   [OP566885, OP566881, PV170748, PV170750, PV170746, PV172328, PV172327, PV170749, PV170751, PV170747]   \n",
       "4                                                                                                     []   \n",
       "5                                                                                             [OR243905]   \n",
       "6                                                        [PP212021, PQ632502, CP173677, JBBEWD000000000]   \n",
       "7                                                                            [PP669793, JBCFYD000000000]   \n",
       "8                                                                               [MF782427, NSKE00000000]   \n",
       "9                                                                                             [OZ026853]   \n",
       "10                                                                           [OR945532, JBFONT000000000]   \n",
       "11                                                [OL504969, PP830069, JBFRYH000000000, JBEHGM000000000]   \n",
       "12                                                [OL504968, PP830068, JBEHGJ000000000, JBEHGL000000000]   \n",
       "13                                                       [OL504562, MN850392, CP172427, JAAKDU000000000]   \n",
       "14                                                [PQ001808, PQ198380, JBFQEG000000000, JBFQEH000000000]   \n",
       "15                                                              [PQ120583, PQ120582, PQ119488, PQ119489]   \n",
       "16                                                              [PQ120583, PQ120582, PQ119488, PQ119489]   \n",
       "17                                                                                                    []   \n",
       "18                                                                                  [LC820896, AP026866]   \n",
       "\n",
       "                                                               Strains  \\\n",
       "0   CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3   \n",
       "1   CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3   \n",
       "2   CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3   \n",
       "3                                               PYCC 9932,  NYNU 22992   \n",
       "4                                                                        \n",
       "5                                     LWZ-6T,  CCAM 1872T,  JCM 39528T   \n",
       "6                                    G13T,  GDMCC 1.4691T,  JCM 37746T   \n",
       "7                                   N2T,  KCTC 102228T,  MCCC 1K08942T   \n",
       "8                                   WN023T,  KCTC 52855T,  ACCC 19978T   \n",
       "9                                  ATCC CRL-1696,  EPC,  ATCC CRL-2872   \n",
       "10                           2305UL40-4T,  KCCM 43505T,  MCCC 1K09164T   \n",
       "11                           ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93   \n",
       "12                          ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99   \n",
       "13                        ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373   \n",
       "14                             H31T,  GDMCC 1.4758T,  JCM 37397T,  H49   \n",
       "15              11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370   \n",
       "16              11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370   \n",
       "17              11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370   \n",
       "18                                                               NT6NT   \n",
       "\n",
       "               Authority                                             DOI  \\\n",
       "0        Wei et al. 2025          https://doi.org/10.1099/ijsem.0.006836   \n",
       "1        Wei et al. 2025          https://doi.org/10.1099/ijsem.0.006836   \n",
       "2        Wei et al. 2025          https://doi.org/10.1099/ijsem.0.006836   \n",
       "3        Gao et al. 2025          https://doi.org/10.1099/ijsem.0.006850   \n",
       "4         Wu et al. 2025          https://doi.org/10.1099/ijsem.0.006839   \n",
       "5         Wu et al. 2025          https://doi.org/10.1099/ijsem.0.006839   \n",
       "6        Gao et al. 2025          https://doi.org/10.1099/ijsem.0.006837   \n",
       "7         He et al. 2025          https://doi.org/10.1099/ijsem.0.006840   \n",
       "8         He et al. 2025          https://doi.org/10.1099/ijsem.0.006840   \n",
       "9    Marquis et al. 2025  https://doi.org/10.6084/m9.figshare.28424588.v   \n",
       "10       Lee et al. 2025          https://doi.org/10.1099/ijsem.0.006834   \n",
       "11        Pu et al. 2025          https://doi.org/10.1099/ijsem.0.006814   \n",
       "12        Pu et al. 2025          https://doi.org/10.1099/ijsem.0.006814   \n",
       "13        Pu et al. 2025          https://doi.org/10.1099/ijsem.0.006814   \n",
       "14        Pu et al. 2025          https://doi.org/10.1099/ijsem.0.006814   \n",
       "15  Klobucar et al. 2025          https://doi.org/10.1099/ijsem.0.006813   \n",
       "16  Klobucar et al. 2025          https://doi.org/10.1099/ijsem.0.006813   \n",
       "17  Klobucar et al. 2025          https://doi.org/10.1099/ijsem.0.006813   \n",
       "18     Huang et al. 2025          https://doi.org/10.1099/ijsem.0.006828   \n",
       "\n",
       "                                filtered_url  \n",
       "0   http://dx.doi.org/10.1099/ijsem.0.006836  \n",
       "1   http://dx.doi.org/10.1099/ijsem.0.006836  \n",
       "2   http://dx.doi.org/10.1099/ijsem.0.006836  \n",
       "3   http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "4   http://dx.doi.org/10.1099/ijsem.0.006839  \n",
       "5   http://dx.doi.org/10.1099/ijsem.0.006839  \n",
       "6   http://dx.doi.org/10.1099/ijsem.0.006837  \n",
       "7   http://dx.doi.org/10.1099/ijsem.0.006840  \n",
       "8   http://dx.doi.org/10.1099/ijsem.0.006840  \n",
       "9   http://dx.doi.org/10.1099/ijsem.0.006753  \n",
       "10  http://dx.doi.org/10.1099/ijsem.0.006834  \n",
       "11  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "12  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "13  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "14  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "15  http://dx.doi.org/10.1099/ijsem.0.006813  \n",
       "16  http://dx.doi.org/10.1099/ijsem.0.006813  \n",
       "17  http://dx.doi.org/10.1099/ijsem.0.006813  \n",
       "18  http://dx.doi.org/10.1099/ijsem.0.006828  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.set_option('max_colwidth', None)\n",
    "pub_df['Strains'] = [', '.join(map(str, l)) for l in pub_df['Strains']]\n",
    "#pub_df['Strains'] = pub_df['Strains'].astype(str) \n",
    "pub_df['Strains'] = pub_df['Strains'].astype(pd.StringDtype())\n",
    "pub_df['Strains'] = pub_df['Strains'].str.replace(',', ', ')\n",
    "pub_df = pub_df.drop_duplicates(subset='PublishedName', keep=\"first\")\n",
    "#try drop duplicate accessions here\n",
    "pub_df.explode(['PublishedName']).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e74e265b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#pub_df['Strains'] = pub_df['Strains'].astype(pd.StringDtype())\n",
    "#pub_df['Strains'] = pub_df['Strains'].str.replace(',', ', ')\n",
    "#pub_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66791f58",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "58034d28",
   "metadata": {},
   "outputs": [],
   "source": [
    "pub2_df = pub_df.explode(['Accessions']).reset_index(drop=True)\n",
    "#pub2_df\n",
    "pub4_df = pub2_df.explode(['PublishedName']).reset_index(drop=True)\n",
    "pub4_df.rename(columns={'Accessions' : 'accession'}, inplace=True)\n",
    "#pub4_df = pub4_df.dropna()\n",
    "#pub4_df = pub4_df.drop_duplicates(subset='accession', keep=\"first\")\n",
    "pub4_df=pub4_df[pub4_df['accession'].isnull() | ~pub4_df[pub4_df['accession'].notnull()].duplicated(subset='accession',keep='first')]\n",
    "#pub4_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4a103350",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_unique= pub4_df.drop_duplicates([\"accession\"], keep=\"first\")\n",
    "#df_unique = df_unique.dropna()\n",
    "#df_unique"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7f80dd1",
   "metadata": {},
   "source": [
    "### Create a dataframe of unique accessions and look up NCBI taxonomy information of each accession with srcchk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84a9da33",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_unique.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8208be48",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_unique['accession'] = df_unique['accession'].astype('str') \n",
    "df_unique.loc[:, 'accession'] = df_unique['accession'].astype('str') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1be92fbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('acclist', 'w') as f:\n",
    "    for text in df_unique['accession'].tolist():\n",
    "        f.write(text + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a97a9150",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error: Malformatted ID \"nan\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.system(\"/netopt/ncbi_tools64/bin/srcchk -i acclist -f taxname,taxid,strain -o acclist.taxdata\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1f0aaf52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>NCBIname</th>\n",
       "      <th>taxid</th>\n",
       "      <th>strain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PQ432548</td>\n",
       "      <td>Rhodotorula sp.</td>\n",
       "      <td>1853554.0</td>\n",
       "      <td>1017-17-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PQ432547</td>\n",
       "      <td>Rhodotorula sp.</td>\n",
       "      <td>1853554.0</td>\n",
       "      <td>1017-17-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OP566885</td>\n",
       "      <td>Farysia sp.</td>\n",
       "      <td>2012362.0</td>\n",
       "      <td>NYNU 22992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OP566881</td>\n",
       "      <td>Farysia sp.</td>\n",
       "      <td>2012362.0</td>\n",
       "      <td>NYNU 22992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PV170748</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>NYNU 22992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PV170750</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>NYNU 22992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PV170746</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>NYNU 22992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PV172328</td>\n",
       "      <td>Farysia sp.</td>\n",
       "      <td>2012362.0</td>\n",
       "      <td>NYNU 229246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PV172327</td>\n",
       "      <td>Farysia sp.</td>\n",
       "      <td>2012362.0</td>\n",
       "      <td>NYNU 229246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PV170749</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>NYNU 229246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PV170751</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>NYNU 229246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PV170747</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>NYNU 229246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>OR243905</td>\n",
       "      <td>Methanosuratincola petrocarbonis</td>\n",
       "      <td>3448697.0</td>\n",
       "      <td>LWZ-6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PP212021</td>\n",
       "      <td>Brachymonas sp.</td>\n",
       "      <td>1936292.0</td>\n",
       "      <td>G13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PQ632502</td>\n",
       "      <td>Brachymonas sp.</td>\n",
       "      <td>1936292.0</td>\n",
       "      <td>G34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CP173677</td>\n",
       "      <td>Brachymonas sp. G13</td>\n",
       "      <td>3130163.0</td>\n",
       "      <td>G13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>JBBEWD000000000</td>\n",
       "      <td>Brachymonas wangyanguii</td>\n",
       "      <td>3130163.0</td>\n",
       "      <td>G34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PP669793</td>\n",
       "      <td>Fodinibius alkaliphilus</td>\n",
       "      <td>3140241.0</td>\n",
       "      <td>N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>JBCFYD000000000</td>\n",
       "      <td>Fodinibius alkaliphilus</td>\n",
       "      <td>3140241.0</td>\n",
       "      <td>N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MF782427</td>\n",
       "      <td>Fodinibius salipaludis</td>\n",
       "      <td>2032627.0</td>\n",
       "      <td>WN023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NSKE00000000</td>\n",
       "      <td>Fodinibius salipaludis</td>\n",
       "      <td>2032627.0</td>\n",
       "      <td>WN023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>OZ026853</td>\n",
       "      <td>Chlamydia vaughanii</td>\n",
       "      <td>3112552.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>OR945532</td>\n",
       "      <td>Aestuariibius violaceus</td>\n",
       "      <td>3234132.0</td>\n",
       "      <td>2305UL40-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>JBFONT000000000</td>\n",
       "      <td>Aestuariibius violaceus</td>\n",
       "      <td>3234132.0</td>\n",
       "      <td>2305UL40-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>OL504969</td>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>3161188.0</td>\n",
       "      <td>ZJ93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PP830069</td>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>3161188.0</td>\n",
       "      <td>ZJ100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>JBFRYH000000000</td>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>3161188.0</td>\n",
       "      <td>ZJ93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>JBEHGM000000000</td>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>3161188.0</td>\n",
       "      <td>ZJ100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>OL504968</td>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>3161189.0</td>\n",
       "      <td>ZJ151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>PP830068</td>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>3161189.0</td>\n",
       "      <td>ZJ99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>JBEHGJ000000000</td>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>3161189.0</td>\n",
       "      <td>ZJ151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>JBEHGL000000000</td>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>3161189.0</td>\n",
       "      <td>ZJ99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>OL504562</td>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>2709400.0</td>\n",
       "      <td>ZJ1593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>MN850392</td>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>2709400.0</td>\n",
       "      <td>ZJ373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>CP172427</td>\n",
       "      <td>Streptococcus sp. ZJ 1593</td>\n",
       "      <td>3366265.0</td>\n",
       "      <td>ZJ 1593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>JAAKDU000000000</td>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>2709400.0</td>\n",
       "      <td>ZJ373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>PQ001808</td>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>3237239.0</td>\n",
       "      <td>H31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>PQ198380</td>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>3237239.0</td>\n",
       "      <td>H49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>JBFQEG000000000</td>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>3237239.0</td>\n",
       "      <td>H31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>JBFQEH000000000</td>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>3237239.0</td>\n",
       "      <td>H49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>PQ120583</td>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>2992041.0</td>\n",
       "      <td>PATHC038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>PQ120582</td>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>2992041.0</td>\n",
       "      <td>HB10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>PQ119488</td>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>2992041.0</td>\n",
       "      <td>PATHC038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PQ119489</td>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>2992041.0</td>\n",
       "      <td>HB10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>LC820896</td>\n",
       "      <td>Oceaniferula spumae</td>\n",
       "      <td>2979115.0</td>\n",
       "      <td>NT6N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>AP026866</td>\n",
       "      <td>Oceaniferula spumae</td>\n",
       "      <td>2979115.0</td>\n",
       "      <td>NT6N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          accession                          NCBIname      taxid       strain\n",
       "0          PQ432548                   Rhodotorula sp.  1853554.0    1017-17-3\n",
       "1          PQ432547                   Rhodotorula sp.  1853554.0    1017-17-3\n",
       "3          OP566885                       Farysia sp.  2012362.0   NYNU 22992\n",
       "4          OP566881                       Farysia sp.  2012362.0   NYNU 22992\n",
       "5          PV170748                    Farysizyma sp.  2650697.0   NYNU 22992\n",
       "6          PV170750                    Farysizyma sp.  2650697.0   NYNU 22992\n",
       "7          PV170746                    Farysizyma sp.  2650697.0   NYNU 22992\n",
       "8          PV172328                       Farysia sp.  2012362.0  NYNU 229246\n",
       "9          PV172327                       Farysia sp.  2012362.0  NYNU 229246\n",
       "10         PV170749                    Farysizyma sp.  2650697.0  NYNU 229246\n",
       "11         PV170751                    Farysizyma sp.  2650697.0  NYNU 229246\n",
       "12         PV170747                    Farysizyma sp.  2650697.0  NYNU 229246\n",
       "13         OR243905  Methanosuratincola petrocarbonis  3448697.0        LWZ-6\n",
       "14         PP212021                   Brachymonas sp.  1936292.0          G13\n",
       "15         PQ632502                   Brachymonas sp.  1936292.0          G34\n",
       "16         CP173677               Brachymonas sp. G13  3130163.0          G13\n",
       "17  JBBEWD000000000           Brachymonas wangyanguii  3130163.0          G34\n",
       "18         PP669793           Fodinibius alkaliphilus  3140241.0           N2\n",
       "19  JBCFYD000000000           Fodinibius alkaliphilus  3140241.0           N2\n",
       "20         MF782427            Fodinibius salipaludis  2032627.0        WN023\n",
       "21     NSKE00000000            Fodinibius salipaludis  2032627.0        WN023\n",
       "22         OZ026853               Chlamydia vaughanii  3112552.0          NaN\n",
       "23         OR945532           Aestuariibius violaceus  3234132.0   2305UL40-4\n",
       "24  JBFONT000000000           Aestuariibius violaceus  3234132.0   2305UL40-4\n",
       "25         OL504969           Streptococcus handemini  3161188.0         ZJ93\n",
       "26         PP830069           Streptococcus handemini  3161188.0        ZJ100\n",
       "27  JBFRYH000000000           Streptococcus handemini  3161188.0         ZJ93\n",
       "28  JBEHGM000000000           Streptococcus handemini  3161188.0        ZJ100\n",
       "29         OL504968       Streptococcus jiangjianxini  3161189.0        ZJ151\n",
       "30         PP830068       Streptococcus jiangjianxini  3161189.0         ZJ99\n",
       "31  JBEHGJ000000000       Streptococcus jiangjianxini  3161189.0        ZJ151\n",
       "32  JBEHGL000000000       Streptococcus jiangjianxini  3161189.0         ZJ99\n",
       "33         OL504562          Streptococcus tangpeifui  2709400.0       ZJ1593\n",
       "34         MN850392          Streptococcus tangpeifui  2709400.0        ZJ373\n",
       "35         CP172427         Streptococcus sp. ZJ 1593  3366265.0      ZJ 1593\n",
       "36  JAAKDU000000000          Streptococcus tangpeifui  2709400.0        ZJ373\n",
       "37         PQ001808       Streptococcus huangxiaojuni  3237239.0          H31\n",
       "38         PQ198380       Streptococcus huangxiaojuni  3237239.0          H49\n",
       "39  JBFQEG000000000       Streptococcus huangxiaojuni  3237239.0          H31\n",
       "40  JBFQEH000000000       Streptococcus huangxiaojuni  3237239.0          H49\n",
       "41         PQ120583              Legionella sheltonii  2992041.0     PATHC038\n",
       "42         PQ120582              Legionella sheltonii  2992041.0         HB10\n",
       "43         PQ119488              Legionella sheltonii  2992041.0     PATHC038\n",
       "44         PQ119489              Legionella sheltonii  2992041.0         HB10\n",
       "45         LC820896               Oceaniferula spumae  2979115.0         NT6N\n",
       "46         AP026866               Oceaniferula spumae  2979115.0         NT6N"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxdata_file_name = (r'acclist.taxdata')    \n",
    "srcchk_df = pd.read_csv(taxdata_file_name, sep='\\t', index_col=None, low_memory=False)\n",
    "srcchk_df.drop(columns=['Unnamed: 4'], inplace=True)\n",
    "srcchk_df.rename(columns={'organism' : 'NCBIname'}, inplace=True)\n",
    "srcchk_df['accession'] = srcchk_df['accession'].astype(str).replace('\\.\\d+', '', regex=True).astype(str)\n",
    "srcchk_df = srcchk_df.dropna(subset=['NCBIname'])\n",
    "srcchk_df "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81b7fe78",
   "metadata": {},
   "source": [
    "### Combine dataframes into one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "7f05118d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PublishedName</th>\n",
       "      <th>NCBIname</th>\n",
       "      <th>Strains</th>\n",
       "      <th>accession</th>\n",
       "      <th>strain</th>\n",
       "      <th>Authority</th>\n",
       "      <th>taxid</th>\n",
       "      <th>DOI</th>\n",
       "      <th>filtered_url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Oceaniferula spumae</td>\n",
       "      <td>Oceaniferula spumae</td>\n",
       "      <td>NT6NT</td>\n",
       "      <td>AP026866</td>\n",
       "      <td>NT6N</td>\n",
       "      <td>Huang et al. 2025</td>\n",
       "      <td>2979115.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006828</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>Streptococcus sp. ZJ 1593</td>\n",
       "      <td>ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373</td>\n",
       "      <td>CP172427</td>\n",
       "      <td>ZJ 1593</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3366265.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Brachymonas wangyanguii</td>\n",
       "      <td>Brachymonas sp. G13</td>\n",
       "      <td>G13T,  GDMCC 1.4691T,  JCM 37746T</td>\n",
       "      <td>CP173677</td>\n",
       "      <td>G13</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>3130163.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006837</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373</td>\n",
       "      <td>JAAKDU000000000</td>\n",
       "      <td>ZJ373</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>2709400.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Brachymonas wangyanguii</td>\n",
       "      <td>Brachymonas wangyanguii</td>\n",
       "      <td>G13T,  GDMCC 1.4691T,  JCM 37746T</td>\n",
       "      <td>JBBEWD000000000</td>\n",
       "      <td>G34</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>3130163.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006837</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Fodinibius alkaliphilus</td>\n",
       "      <td>Fodinibius alkaliphilus</td>\n",
       "      <td>N2T,  KCTC 102228T,  MCCC 1K08942T</td>\n",
       "      <td>JBCFYD000000000</td>\n",
       "      <td>N2</td>\n",
       "      <td>He et al. 2025</td>\n",
       "      <td>3140241.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99</td>\n",
       "      <td>JBEHGJ000000000</td>\n",
       "      <td>ZJ151</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3161189.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99</td>\n",
       "      <td>JBEHGL000000000</td>\n",
       "      <td>ZJ99</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3161189.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93</td>\n",
       "      <td>JBEHGM000000000</td>\n",
       "      <td>ZJ100</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3161188.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Aestuariibius violaceus</td>\n",
       "      <td>Aestuariibius violaceus</td>\n",
       "      <td>2305UL40-4T,  KCCM 43505T,  MCCC 1K09164T</td>\n",
       "      <td>JBFONT000000000</td>\n",
       "      <td>2305UL40-4</td>\n",
       "      <td>Lee et al. 2025</td>\n",
       "      <td>3234132.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006834</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>H31T,  GDMCC 1.4758T,  JCM 37397T,  H49</td>\n",
       "      <td>JBFQEG000000000</td>\n",
       "      <td>H31</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3237239.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>H31T,  GDMCC 1.4758T,  JCM 37397T,  H49</td>\n",
       "      <td>JBFQEH000000000</td>\n",
       "      <td>H49</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3237239.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93</td>\n",
       "      <td>JBFRYH000000000</td>\n",
       "      <td>ZJ93</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3161188.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Oceaniferula spumae</td>\n",
       "      <td>Oceaniferula spumae</td>\n",
       "      <td>NT6NT</td>\n",
       "      <td>LC820896</td>\n",
       "      <td>NT6N</td>\n",
       "      <td>Huang et al. 2025</td>\n",
       "      <td>2979115.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006828</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Fodinibius salipaludis</td>\n",
       "      <td>Fodinibius salipaludis</td>\n",
       "      <td>WN023T,  KCTC 52855T,  ACCC 19978T</td>\n",
       "      <td>MF782427</td>\n",
       "      <td>WN023</td>\n",
       "      <td>He et al. 2025</td>\n",
       "      <td>2032627.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373</td>\n",
       "      <td>MN850392</td>\n",
       "      <td>ZJ373</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>2709400.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Fodinibius salipaludis</td>\n",
       "      <td>Fodinibius salipaludis</td>\n",
       "      <td>WN023T,  KCTC 52855T,  ACCC 19978T</td>\n",
       "      <td>NSKE00000000</td>\n",
       "      <td>WN023</td>\n",
       "      <td>He et al. 2025</td>\n",
       "      <td>2032627.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>Streptococcus tangpeifui</td>\n",
       "      <td>ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373</td>\n",
       "      <td>OL504562</td>\n",
       "      <td>ZJ1593</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>2709400.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99</td>\n",
       "      <td>OL504968</td>\n",
       "      <td>ZJ151</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3161189.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93</td>\n",
       "      <td>OL504969</td>\n",
       "      <td>ZJ93</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3161188.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysia sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>OP566881</td>\n",
       "      <td>NYNU 22992</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2012362.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysia sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>OP566885</td>\n",
       "      <td>NYNU 22992</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2012362.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Methanosuratincola petrocarbonis</td>\n",
       "      <td>Methanosuratincola petrocarbonis</td>\n",
       "      <td>LWZ-6T,  CCAM 1872T,  JCM 39528T</td>\n",
       "      <td>OR243905</td>\n",
       "      <td>LWZ-6</td>\n",
       "      <td>Wu et al. 2025</td>\n",
       "      <td>3448697.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006839</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Aestuariibius violaceus</td>\n",
       "      <td>Aestuariibius violaceus</td>\n",
       "      <td>2305UL40-4T,  KCCM 43505T,  MCCC 1K09164T</td>\n",
       "      <td>OR945532</td>\n",
       "      <td>2305UL40-4</td>\n",
       "      <td>Lee et al. 2025</td>\n",
       "      <td>3234132.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006834</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Chlamydia vaughanii</td>\n",
       "      <td>Chlamydia vaughanii</td>\n",
       "      <td>ATCC CRL-1696,  EPC,  ATCC CRL-2872</td>\n",
       "      <td>OZ026853</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Marquis et al. 2025</td>\n",
       "      <td>3112552.0</td>\n",
       "      <td>https://doi.org/10.6084/m9.figshare.28424588.v</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Brachymonas wangyanguii</td>\n",
       "      <td>Brachymonas sp.</td>\n",
       "      <td>G13T,  GDMCC 1.4691T,  JCM 37746T</td>\n",
       "      <td>PP212021</td>\n",
       "      <td>G13</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>1936292.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006837</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Fodinibius alkaliphilus</td>\n",
       "      <td>Fodinibius alkaliphilus</td>\n",
       "      <td>N2T,  KCTC 102228T,  MCCC 1K08942T</td>\n",
       "      <td>PP669793</td>\n",
       "      <td>N2</td>\n",
       "      <td>He et al. 2025</td>\n",
       "      <td>3140241.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>Streptococcus jiangjianxini</td>\n",
       "      <td>ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99</td>\n",
       "      <td>PP830068</td>\n",
       "      <td>ZJ99</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3161189.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>Streptococcus handemini</td>\n",
       "      <td>ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93</td>\n",
       "      <td>PP830069</td>\n",
       "      <td>ZJ100</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3161188.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>H31T,  GDMCC 1.4758T,  JCM 37397T,  H49</td>\n",
       "      <td>PQ001808</td>\n",
       "      <td>H31</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3237239.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td>PQ119488</td>\n",
       "      <td>PATHC038</td>\n",
       "      <td>Klobucar et al. 2025</td>\n",
       "      <td>2992041.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td>PQ119489</td>\n",
       "      <td>HB10</td>\n",
       "      <td>Klobucar et al. 2025</td>\n",
       "      <td>2992041.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td>PQ120582</td>\n",
       "      <td>HB10</td>\n",
       "      <td>Klobucar et al. 2025</td>\n",
       "      <td>2992041.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td>PQ120583</td>\n",
       "      <td>PATHC038</td>\n",
       "      <td>Klobucar et al. 2025</td>\n",
       "      <td>2992041.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>Streptococcus huangxiaojuni</td>\n",
       "      <td>H31T,  GDMCC 1.4758T,  JCM 37397T,  H49</td>\n",
       "      <td>PQ198380</td>\n",
       "      <td>H49</td>\n",
       "      <td>Pu et al. 2025</td>\n",
       "      <td>3237239.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>F.Y. Bai</td>\n",
       "      <td>Rhodotorula sp.</td>\n",
       "      <td>CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3</td>\n",
       "      <td>PQ432547</td>\n",
       "      <td>1017-17-3</td>\n",
       "      <td>Wei et al. 2025</td>\n",
       "      <td>1853554.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006836</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>F.Y. Bai</td>\n",
       "      <td>Rhodotorula sp.</td>\n",
       "      <td>CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3</td>\n",
       "      <td>PQ432548</td>\n",
       "      <td>1017-17-3</td>\n",
       "      <td>Wei et al. 2025</td>\n",
       "      <td>1853554.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006836</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Brachymonas wangyanguii</td>\n",
       "      <td>Brachymonas sp.</td>\n",
       "      <td>G13T,  GDMCC 1.4691T,  JCM 37746T</td>\n",
       "      <td>PQ632502</td>\n",
       "      <td>G34</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>1936292.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006837</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>PV170746</td>\n",
       "      <td>NYNU 22992</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>PV170747</td>\n",
       "      <td>NYNU 229246</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>PV170748</td>\n",
       "      <td>NYNU 22992</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>PV170749</td>\n",
       "      <td>NYNU 229246</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>PV170750</td>\n",
       "      <td>NYNU 22992</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysizyma sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>PV170751</td>\n",
       "      <td>NYNU 229246</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2650697.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysia sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>PV172327</td>\n",
       "      <td>NYNU 229246</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2012362.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Farysia photiniae</td>\n",
       "      <td>Farysia sp.</td>\n",
       "      <td>PYCC 9932,  NYNU 22992</td>\n",
       "      <td>PV172328</td>\n",
       "      <td>NYNU 229246</td>\n",
       "      <td>Gao et al. 2025</td>\n",
       "      <td>2012362.0</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>F.Y. Bai</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wei et al. 2025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006836</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wu et al. 2025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006839</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Legionella sheltonii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Klobucar et al. 2025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td>http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        PublishedName                          NCBIname  \\\n",
       "0                Oceaniferula spumae                Oceaniferula spumae   \n",
       "1           Streptococcus tangpeifui          Streptococcus sp. ZJ 1593   \n",
       "2            Brachymonas wangyanguii                Brachymonas sp. G13   \n",
       "3           Streptococcus tangpeifui           Streptococcus tangpeifui   \n",
       "4            Brachymonas wangyanguii            Brachymonas wangyanguii   \n",
       "5            Fodinibius alkaliphilus            Fodinibius alkaliphilus   \n",
       "6        Streptococcus jiangjianxini        Streptococcus jiangjianxini   \n",
       "7        Streptococcus jiangjianxini        Streptococcus jiangjianxini   \n",
       "8            Streptococcus handemini            Streptococcus handemini   \n",
       "9            Aestuariibius violaceus            Aestuariibius violaceus   \n",
       "10       Streptococcus huangxiaojuni        Streptococcus huangxiaojuni   \n",
       "11       Streptococcus huangxiaojuni        Streptococcus huangxiaojuni   \n",
       "12           Streptococcus handemini            Streptococcus handemini   \n",
       "13               Oceaniferula spumae                Oceaniferula spumae   \n",
       "14            Fodinibius salipaludis             Fodinibius salipaludis   \n",
       "15          Streptococcus tangpeifui           Streptococcus tangpeifui   \n",
       "16            Fodinibius salipaludis             Fodinibius salipaludis   \n",
       "17          Streptococcus tangpeifui           Streptococcus tangpeifui   \n",
       "18       Streptococcus jiangjianxini        Streptococcus jiangjianxini   \n",
       "19           Streptococcus handemini            Streptococcus handemini   \n",
       "20                 Farysia photiniae                        Farysia sp.   \n",
       "21                 Farysia photiniae                        Farysia sp.   \n",
       "22  Methanosuratincola petrocarbonis   Methanosuratincola petrocarbonis   \n",
       "23           Aestuariibius violaceus            Aestuariibius violaceus   \n",
       "24               Chlamydia vaughanii                Chlamydia vaughanii   \n",
       "25           Brachymonas wangyanguii                    Brachymonas sp.   \n",
       "26           Fodinibius alkaliphilus            Fodinibius alkaliphilus   \n",
       "27       Streptococcus jiangjianxini        Streptococcus jiangjianxini   \n",
       "28           Streptococcus handemini            Streptococcus handemini   \n",
       "29       Streptococcus huangxiaojuni        Streptococcus huangxiaojuni   \n",
       "30              Legionella sheltonii               Legionella sheltonii   \n",
       "31              Legionella sheltonii               Legionella sheltonii   \n",
       "32              Legionella sheltonii               Legionella sheltonii   \n",
       "33              Legionella sheltonii               Legionella sheltonii   \n",
       "34       Streptococcus huangxiaojuni        Streptococcus huangxiaojuni   \n",
       "35                          F.Y. Bai                    Rhodotorula sp.   \n",
       "36                          F.Y. Bai                    Rhodotorula sp.   \n",
       "37           Brachymonas wangyanguii                    Brachymonas sp.   \n",
       "38                 Farysia photiniae                     Farysizyma sp.   \n",
       "39                 Farysia photiniae                     Farysizyma sp.   \n",
       "40                 Farysia photiniae                     Farysizyma sp.   \n",
       "41                 Farysia photiniae                     Farysizyma sp.   \n",
       "42                 Farysia photiniae                     Farysizyma sp.   \n",
       "43                 Farysia photiniae                     Farysizyma sp.   \n",
       "44                 Farysia photiniae                        Farysia sp.   \n",
       "45                 Farysia photiniae                        Farysia sp.   \n",
       "46                          F.Y. Bai                                NaN   \n",
       "47                                NaN                               NaN   \n",
       "48              Legionella sheltonii                                NaN   \n",
       "\n",
       "                                                               Strains  \\\n",
       "0                                                                NT6NT   \n",
       "1                         ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373   \n",
       "2                                    G13T,  GDMCC 1.4691T,  JCM 37746T   \n",
       "3                         ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373   \n",
       "4                                    G13T,  GDMCC 1.4691T,  JCM 37746T   \n",
       "5                                   N2T,  KCTC 102228T,  MCCC 1K08942T   \n",
       "6                           ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99   \n",
       "7                           ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99   \n",
       "8                            ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93   \n",
       "9                            2305UL40-4T,  KCCM 43505T,  MCCC 1K09164T   \n",
       "10                             H31T,  GDMCC 1.4758T,  JCM 37397T,  H49   \n",
       "11                             H31T,  GDMCC 1.4758T,  JCM 37397T,  H49   \n",
       "12                           ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93   \n",
       "13                                                               NT6NT   \n",
       "14                                  WN023T,  KCTC 52855T,  ACCC 19978T   \n",
       "15                        ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373   \n",
       "16                                  WN023T,  KCTC 52855T,  ACCC 19978T   \n",
       "17                        ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373   \n",
       "18                          ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99   \n",
       "19                           ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93   \n",
       "20                                              PYCC 9932,  NYNU 22992   \n",
       "21                                              PYCC 9932,  NYNU 22992   \n",
       "22                                    LWZ-6T,  CCAM 1872T,  JCM 39528T   \n",
       "23                           2305UL40-4T,  KCCM 43505T,  MCCC 1K09164T   \n",
       "24                                 ATCC CRL-1696,  EPC,  ATCC CRL-2872   \n",
       "25                                   G13T,  GDMCC 1.4691T,  JCM 37746T   \n",
       "26                                  N2T,  KCTC 102228T,  MCCC 1K08942T   \n",
       "27                          ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99   \n",
       "28                           ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93   \n",
       "29                             H31T,  GDMCC 1.4758T,  JCM 37397T,  H49   \n",
       "30              11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370   \n",
       "31              11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370   \n",
       "32              11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370   \n",
       "33              11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370   \n",
       "34                             H31T,  GDMCC 1.4758T,  JCM 37397T,  H49   \n",
       "35  CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3   \n",
       "36  CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3   \n",
       "37                                   G13T,  GDMCC 1.4691T,  JCM 37746T   \n",
       "38                                              PYCC 9932,  NYNU 22992   \n",
       "39                                              PYCC 9932,  NYNU 22992   \n",
       "40                                              PYCC 9932,  NYNU 22992   \n",
       "41                                              PYCC 9932,  NYNU 22992   \n",
       "42                                              PYCC 9932,  NYNU 22992   \n",
       "43                                              PYCC 9932,  NYNU 22992   \n",
       "44                                              PYCC 9932,  NYNU 22992   \n",
       "45                                              PYCC 9932,  NYNU 22992   \n",
       "46  CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3   \n",
       "47                                                                       \n",
       "48              11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370   \n",
       "\n",
       "          accession       strain             Authority      taxid  \\\n",
       "0          AP026866         NT6N     Huang et al. 2025  2979115.0   \n",
       "1          CP172427      ZJ 1593        Pu et al. 2025  3366265.0   \n",
       "2          CP173677          G13       Gao et al. 2025  3130163.0   \n",
       "3   JAAKDU000000000        ZJ373        Pu et al. 2025  2709400.0   \n",
       "4   JBBEWD000000000          G34       Gao et al. 2025  3130163.0   \n",
       "5   JBCFYD000000000           N2        He et al. 2025  3140241.0   \n",
       "6   JBEHGJ000000000        ZJ151        Pu et al. 2025  3161189.0   \n",
       "7   JBEHGL000000000         ZJ99        Pu et al. 2025  3161189.0   \n",
       "8   JBEHGM000000000        ZJ100        Pu et al. 2025  3161188.0   \n",
       "9   JBFONT000000000   2305UL40-4       Lee et al. 2025  3234132.0   \n",
       "10  JBFQEG000000000          H31        Pu et al. 2025  3237239.0   \n",
       "11  JBFQEH000000000          H49        Pu et al. 2025  3237239.0   \n",
       "12  JBFRYH000000000         ZJ93        Pu et al. 2025  3161188.0   \n",
       "13         LC820896         NT6N     Huang et al. 2025  2979115.0   \n",
       "14         MF782427        WN023        He et al. 2025  2032627.0   \n",
       "15         MN850392        ZJ373        Pu et al. 2025  2709400.0   \n",
       "16     NSKE00000000        WN023        He et al. 2025  2032627.0   \n",
       "17         OL504562       ZJ1593        Pu et al. 2025  2709400.0   \n",
       "18         OL504968        ZJ151        Pu et al. 2025  3161189.0   \n",
       "19         OL504969         ZJ93        Pu et al. 2025  3161188.0   \n",
       "20         OP566881   NYNU 22992       Gao et al. 2025  2012362.0   \n",
       "21         OP566885   NYNU 22992       Gao et al. 2025  2012362.0   \n",
       "22         OR243905        LWZ-6        Wu et al. 2025  3448697.0   \n",
       "23         OR945532   2305UL40-4       Lee et al. 2025  3234132.0   \n",
       "24         OZ026853          NaN   Marquis et al. 2025  3112552.0   \n",
       "25         PP212021          G13       Gao et al. 2025  1936292.0   \n",
       "26         PP669793           N2        He et al. 2025  3140241.0   \n",
       "27         PP830068         ZJ99        Pu et al. 2025  3161189.0   \n",
       "28         PP830069        ZJ100        Pu et al. 2025  3161188.0   \n",
       "29         PQ001808          H31        Pu et al. 2025  3237239.0   \n",
       "30         PQ119488     PATHC038  Klobucar et al. 2025  2992041.0   \n",
       "31         PQ119489         HB10  Klobucar et al. 2025  2992041.0   \n",
       "32         PQ120582         HB10  Klobucar et al. 2025  2992041.0   \n",
       "33         PQ120583     PATHC038  Klobucar et al. 2025  2992041.0   \n",
       "34         PQ198380          H49        Pu et al. 2025  3237239.0   \n",
       "35         PQ432547    1017-17-3       Wei et al. 2025  1853554.0   \n",
       "36         PQ432548    1017-17-3       Wei et al. 2025  1853554.0   \n",
       "37         PQ632502          G34       Gao et al. 2025  1936292.0   \n",
       "38         PV170746   NYNU 22992       Gao et al. 2025  2650697.0   \n",
       "39         PV170747  NYNU 229246       Gao et al. 2025  2650697.0   \n",
       "40         PV170748   NYNU 22992       Gao et al. 2025  2650697.0   \n",
       "41         PV170749  NYNU 229246       Gao et al. 2025  2650697.0   \n",
       "42         PV170750   NYNU 22992       Gao et al. 2025  2650697.0   \n",
       "43         PV170751  NYNU 229246       Gao et al. 2025  2650697.0   \n",
       "44         PV172327  NYNU 229246       Gao et al. 2025  2012362.0   \n",
       "45         PV172328  NYNU 229246       Gao et al. 2025  2012362.0   \n",
       "46              NaN          NaN       Wei et al. 2025        NaN   \n",
       "47              NaN          NaN        Wu et al. 2025        NaN   \n",
       "48              NaN          NaN  Klobucar et al. 2025        NaN   \n",
       "\n",
       "                                               DOI  \\\n",
       "0           https://doi.org/10.1099/ijsem.0.006828   \n",
       "1           https://doi.org/10.1099/ijsem.0.006814   \n",
       "2           https://doi.org/10.1099/ijsem.0.006837   \n",
       "3           https://doi.org/10.1099/ijsem.0.006814   \n",
       "4           https://doi.org/10.1099/ijsem.0.006837   \n",
       "5           https://doi.org/10.1099/ijsem.0.006840   \n",
       "6           https://doi.org/10.1099/ijsem.0.006814   \n",
       "7           https://doi.org/10.1099/ijsem.0.006814   \n",
       "8           https://doi.org/10.1099/ijsem.0.006814   \n",
       "9           https://doi.org/10.1099/ijsem.0.006834   \n",
       "10          https://doi.org/10.1099/ijsem.0.006814   \n",
       "11          https://doi.org/10.1099/ijsem.0.006814   \n",
       "12          https://doi.org/10.1099/ijsem.0.006814   \n",
       "13          https://doi.org/10.1099/ijsem.0.006828   \n",
       "14          https://doi.org/10.1099/ijsem.0.006840   \n",
       "15          https://doi.org/10.1099/ijsem.0.006814   \n",
       "16          https://doi.org/10.1099/ijsem.0.006840   \n",
       "17          https://doi.org/10.1099/ijsem.0.006814   \n",
       "18          https://doi.org/10.1099/ijsem.0.006814   \n",
       "19          https://doi.org/10.1099/ijsem.0.006814   \n",
       "20          https://doi.org/10.1099/ijsem.0.006850   \n",
       "21          https://doi.org/10.1099/ijsem.0.006850   \n",
       "22          https://doi.org/10.1099/ijsem.0.006839   \n",
       "23          https://doi.org/10.1099/ijsem.0.006834   \n",
       "24  https://doi.org/10.6084/m9.figshare.28424588.v   \n",
       "25          https://doi.org/10.1099/ijsem.0.006837   \n",
       "26          https://doi.org/10.1099/ijsem.0.006840   \n",
       "27          https://doi.org/10.1099/ijsem.0.006814   \n",
       "28          https://doi.org/10.1099/ijsem.0.006814   \n",
       "29          https://doi.org/10.1099/ijsem.0.006814   \n",
       "30          https://doi.org/10.1099/ijsem.0.006813   \n",
       "31          https://doi.org/10.1099/ijsem.0.006813   \n",
       "32          https://doi.org/10.1099/ijsem.0.006813   \n",
       "33          https://doi.org/10.1099/ijsem.0.006813   \n",
       "34          https://doi.org/10.1099/ijsem.0.006814   \n",
       "35          https://doi.org/10.1099/ijsem.0.006836   \n",
       "36          https://doi.org/10.1099/ijsem.0.006836   \n",
       "37          https://doi.org/10.1099/ijsem.0.006837   \n",
       "38          https://doi.org/10.1099/ijsem.0.006850   \n",
       "39          https://doi.org/10.1099/ijsem.0.006850   \n",
       "40          https://doi.org/10.1099/ijsem.0.006850   \n",
       "41          https://doi.org/10.1099/ijsem.0.006850   \n",
       "42          https://doi.org/10.1099/ijsem.0.006850   \n",
       "43          https://doi.org/10.1099/ijsem.0.006850   \n",
       "44          https://doi.org/10.1099/ijsem.0.006850   \n",
       "45          https://doi.org/10.1099/ijsem.0.006850   \n",
       "46          https://doi.org/10.1099/ijsem.0.006836   \n",
       "47          https://doi.org/10.1099/ijsem.0.006839   \n",
       "48          https://doi.org/10.1099/ijsem.0.006813   \n",
       "\n",
       "                                filtered_url  \n",
       "0   http://dx.doi.org/10.1099/ijsem.0.006828  \n",
       "1   http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "2   http://dx.doi.org/10.1099/ijsem.0.006837  \n",
       "3   http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "4   http://dx.doi.org/10.1099/ijsem.0.006837  \n",
       "5   http://dx.doi.org/10.1099/ijsem.0.006840  \n",
       "6   http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "7   http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "8   http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "9   http://dx.doi.org/10.1099/ijsem.0.006834  \n",
       "10  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "11  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "12  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "13  http://dx.doi.org/10.1099/ijsem.0.006828  \n",
       "14  http://dx.doi.org/10.1099/ijsem.0.006840  \n",
       "15  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "16  http://dx.doi.org/10.1099/ijsem.0.006840  \n",
       "17  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "18  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "19  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "20  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "21  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "22  http://dx.doi.org/10.1099/ijsem.0.006839  \n",
       "23  http://dx.doi.org/10.1099/ijsem.0.006834  \n",
       "24  http://dx.doi.org/10.1099/ijsem.0.006753  \n",
       "25  http://dx.doi.org/10.1099/ijsem.0.006837  \n",
       "26  http://dx.doi.org/10.1099/ijsem.0.006840  \n",
       "27  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "28  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "29  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "30  http://dx.doi.org/10.1099/ijsem.0.006813  \n",
       "31  http://dx.doi.org/10.1099/ijsem.0.006813  \n",
       "32  http://dx.doi.org/10.1099/ijsem.0.006813  \n",
       "33  http://dx.doi.org/10.1099/ijsem.0.006813  \n",
       "34  http://dx.doi.org/10.1099/ijsem.0.006814  \n",
       "35  http://dx.doi.org/10.1099/ijsem.0.006836  \n",
       "36  http://dx.doi.org/10.1099/ijsem.0.006836  \n",
       "37  http://dx.doi.org/10.1099/ijsem.0.006837  \n",
       "38  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "39  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "40  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "41  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "42  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "43  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "44  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "45  http://dx.doi.org/10.1099/ijsem.0.006850  \n",
       "46  http://dx.doi.org/10.1099/ijsem.0.006836  \n",
       "47  http://dx.doi.org/10.1099/ijsem.0.006839  \n",
       "48  http://dx.doi.org/10.1099/ijsem.0.006813  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combine_df=pd.merge(left=pub4_df, right=srcchk_df, left_on='accession', right_on='accession', how = 'outer')\n",
    "combine_df = combine_df[['PublishedName', 'NCBIname', 'Strains', 'accession', 'strain', 'Authority', 'taxid', 'DOI', 'filtered_url' ]]\n",
    "combine_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "8d07fbae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_2c4c2_row0_col0, #T_2c4c2_row0_col1, #T_2c4c2_row1_col0, #T_2c4c2_row1_col1, #T_2c4c2_row2_col0, #T_2c4c2_row2_col1, #T_2c4c2_row3_col0, #T_2c4c2_row3_col1, #T_2c4c2_row4_col0, #T_2c4c2_row4_col1, #T_2c4c2_row5_col0, #T_2c4c2_row5_col1, #T_2c4c2_row6_col0, #T_2c4c2_row6_col1, #T_2c4c2_row7_col0, #T_2c4c2_row7_col1, #T_2c4c2_row8_col0, #T_2c4c2_row8_col1, #T_2c4c2_row9_col0, #T_2c4c2_row9_col1, #T_2c4c2_row10_col0, #T_2c4c2_row10_col1, #T_2c4c2_row11_col0, #T_2c4c2_row11_col1, #T_2c4c2_row12_col0, #T_2c4c2_row12_col1, #T_2c4c2_row13_col0, #T_2c4c2_row13_col1, #T_2c4c2_row14_col0, #T_2c4c2_row14_col1, #T_2c4c2_row15_col0, #T_2c4c2_row15_col1, #T_2c4c2_row16_col0, #T_2c4c2_row16_col1, #T_2c4c2_row17_col0, #T_2c4c2_row17_col1, #T_2c4c2_row18_col0, #T_2c4c2_row18_col1, #T_2c4c2_row19_col0, #T_2c4c2_row19_col1, #T_2c4c2_row20_col0, #T_2c4c2_row20_col1, #T_2c4c2_row21_col0, #T_2c4c2_row21_col1, #T_2c4c2_row22_col0, #T_2c4c2_row22_col1, #T_2c4c2_row23_col0, #T_2c4c2_row23_col1, #T_2c4c2_row24_col0, #T_2c4c2_row24_col1, #T_2c4c2_row25_col0, #T_2c4c2_row25_col1, #T_2c4c2_row26_col0, #T_2c4c2_row26_col1, #T_2c4c2_row27_col0, #T_2c4c2_row27_col1, #T_2c4c2_row28_col0, #T_2c4c2_row28_col1, #T_2c4c2_row29_col0, #T_2c4c2_row29_col1, #T_2c4c2_row30_col0, #T_2c4c2_row30_col1, #T_2c4c2_row31_col0, #T_2c4c2_row31_col1, #T_2c4c2_row32_col0, #T_2c4c2_row32_col1, #T_2c4c2_row33_col0, #T_2c4c2_row33_col1, #T_2c4c2_row34_col0, #T_2c4c2_row34_col1, #T_2c4c2_row35_col0, #T_2c4c2_row35_col1, #T_2c4c2_row36_col0, #T_2c4c2_row36_col1, #T_2c4c2_row37_col0, #T_2c4c2_row37_col1, #T_2c4c2_row38_col0, #T_2c4c2_row38_col1, #T_2c4c2_row39_col0, #T_2c4c2_row39_col1, #T_2c4c2_row40_col0, #T_2c4c2_row40_col1, #T_2c4c2_row41_col0, #T_2c4c2_row41_col1, #T_2c4c2_row42_col0, #T_2c4c2_row42_col1, #T_2c4c2_row43_col0, #T_2c4c2_row43_col1, #T_2c4c2_row44_col0, #T_2c4c2_row44_col1, #T_2c4c2_row45_col0, #T_2c4c2_row45_col1, #T_2c4c2_row46_col0, #T_2c4c2_row46_col1, #T_2c4c2_row47_col0, #T_2c4c2_row47_col1, #T_2c4c2_row48_col0, #T_2c4c2_row48_col1 {\n",
       "  background-color: #FFB3BA;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_2c4c2\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_2c4c2_level0_col0\" class=\"col_heading level0 col0\" >PublishedName</th>\n",
       "      <th id=\"T_2c4c2_level0_col1\" class=\"col_heading level0 col1\" >NCBIname</th>\n",
       "      <th id=\"T_2c4c2_level0_col2\" class=\"col_heading level0 col2\" >Strains</th>\n",
       "      <th id=\"T_2c4c2_level0_col3\" class=\"col_heading level0 col3\" >accession</th>\n",
       "      <th id=\"T_2c4c2_level0_col4\" class=\"col_heading level0 col4\" >strain</th>\n",
       "      <th id=\"T_2c4c2_level0_col5\" class=\"col_heading level0 col5\" >Authority</th>\n",
       "      <th id=\"T_2c4c2_level0_col6\" class=\"col_heading level0 col6\" >taxid</th>\n",
       "      <th id=\"T_2c4c2_level0_col7\" class=\"col_heading level0 col7\" >DOI</th>\n",
       "      <th id=\"T_2c4c2_level0_col8\" class=\"col_heading level0 col8\" >filtered_url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_2c4c2_row0_col0\" class=\"data row0 col0\" >Oceaniferula spumae </td>\n",
       "      <td id=\"T_2c4c2_row0_col1\" class=\"data row0 col1\" >Oceaniferula spumae</td>\n",
       "      <td id=\"T_2c4c2_row0_col2\" class=\"data row0 col2\" >NT6NT</td>\n",
       "      <td id=\"T_2c4c2_row0_col3\" class=\"data row0 col3\" >AP026866</td>\n",
       "      <td id=\"T_2c4c2_row0_col4\" class=\"data row0 col4\" >NT6N</td>\n",
       "      <td id=\"T_2c4c2_row0_col5\" class=\"data row0 col5\" >Huang et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row0_col6\" class=\"data row0 col6\" >2979115.000000</td>\n",
       "      <td id=\"T_2c4c2_row0_col7\" class=\"data row0 col7\" >https://doi.org/10.1099/ijsem.0.006828</td>\n",
       "      <td id=\"T_2c4c2_row0_col8\" class=\"data row0 col8\" >http://dx.doi.org/10.1099/ijsem.0.006828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_2c4c2_row1_col0\" class=\"data row1 col0\" >Streptococcus tangpeifui </td>\n",
       "      <td id=\"T_2c4c2_row1_col1\" class=\"data row1 col1\" >Streptococcus sp. ZJ 1593</td>\n",
       "      <td id=\"T_2c4c2_row1_col2\" class=\"data row1 col2\" >ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373</td>\n",
       "      <td id=\"T_2c4c2_row1_col3\" class=\"data row1 col3\" >CP172427</td>\n",
       "      <td id=\"T_2c4c2_row1_col4\" class=\"data row1 col4\" >ZJ 1593</td>\n",
       "      <td id=\"T_2c4c2_row1_col5\" class=\"data row1 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row1_col6\" class=\"data row1 col6\" >3366265.000000</td>\n",
       "      <td id=\"T_2c4c2_row1_col7\" class=\"data row1 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row1_col8\" class=\"data row1 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_2c4c2_row2_col0\" class=\"data row2 col0\" >Brachymonas wangyanguii </td>\n",
       "      <td id=\"T_2c4c2_row2_col1\" class=\"data row2 col1\" >Brachymonas sp. G13</td>\n",
       "      <td id=\"T_2c4c2_row2_col2\" class=\"data row2 col2\" >G13T,  GDMCC 1.4691T,  JCM 37746T</td>\n",
       "      <td id=\"T_2c4c2_row2_col3\" class=\"data row2 col3\" >CP173677</td>\n",
       "      <td id=\"T_2c4c2_row2_col4\" class=\"data row2 col4\" >G13</td>\n",
       "      <td id=\"T_2c4c2_row2_col5\" class=\"data row2 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row2_col6\" class=\"data row2 col6\" >3130163.000000</td>\n",
       "      <td id=\"T_2c4c2_row2_col7\" class=\"data row2 col7\" >https://doi.org/10.1099/ijsem.0.006837</td>\n",
       "      <td id=\"T_2c4c2_row2_col8\" class=\"data row2 col8\" >http://dx.doi.org/10.1099/ijsem.0.006837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_2c4c2_row3_col0\" class=\"data row3 col0\" >Streptococcus tangpeifui </td>\n",
       "      <td id=\"T_2c4c2_row3_col1\" class=\"data row3 col1\" >Streptococcus tangpeifui</td>\n",
       "      <td id=\"T_2c4c2_row3_col2\" class=\"data row3 col2\" >ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373</td>\n",
       "      <td id=\"T_2c4c2_row3_col3\" class=\"data row3 col3\" >JAAKDU000000000</td>\n",
       "      <td id=\"T_2c4c2_row3_col4\" class=\"data row3 col4\" >ZJ373</td>\n",
       "      <td id=\"T_2c4c2_row3_col5\" class=\"data row3 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row3_col6\" class=\"data row3 col6\" >2709400.000000</td>\n",
       "      <td id=\"T_2c4c2_row3_col7\" class=\"data row3 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row3_col8\" class=\"data row3 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_2c4c2_row4_col0\" class=\"data row4 col0\" >Brachymonas wangyanguii </td>\n",
       "      <td id=\"T_2c4c2_row4_col1\" class=\"data row4 col1\" >Brachymonas wangyanguii</td>\n",
       "      <td id=\"T_2c4c2_row4_col2\" class=\"data row4 col2\" >G13T,  GDMCC 1.4691T,  JCM 37746T</td>\n",
       "      <td id=\"T_2c4c2_row4_col3\" class=\"data row4 col3\" >JBBEWD000000000</td>\n",
       "      <td id=\"T_2c4c2_row4_col4\" class=\"data row4 col4\" >G34</td>\n",
       "      <td id=\"T_2c4c2_row4_col5\" class=\"data row4 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row4_col6\" class=\"data row4 col6\" >3130163.000000</td>\n",
       "      <td id=\"T_2c4c2_row4_col7\" class=\"data row4 col7\" >https://doi.org/10.1099/ijsem.0.006837</td>\n",
       "      <td id=\"T_2c4c2_row4_col8\" class=\"data row4 col8\" >http://dx.doi.org/10.1099/ijsem.0.006837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_2c4c2_row5_col0\" class=\"data row5 col0\" >Fodinibius alkaliphilus </td>\n",
       "      <td id=\"T_2c4c2_row5_col1\" class=\"data row5 col1\" >Fodinibius alkaliphilus</td>\n",
       "      <td id=\"T_2c4c2_row5_col2\" class=\"data row5 col2\" >N2T,  KCTC 102228T,  MCCC 1K08942T</td>\n",
       "      <td id=\"T_2c4c2_row5_col3\" class=\"data row5 col3\" >JBCFYD000000000</td>\n",
       "      <td id=\"T_2c4c2_row5_col4\" class=\"data row5 col4\" >N2</td>\n",
       "      <td id=\"T_2c4c2_row5_col5\" class=\"data row5 col5\" >He et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row5_col6\" class=\"data row5 col6\" >3140241.000000</td>\n",
       "      <td id=\"T_2c4c2_row5_col7\" class=\"data row5 col7\" >https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td id=\"T_2c4c2_row5_col8\" class=\"data row5 col8\" >http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_2c4c2_row6_col0\" class=\"data row6 col0\" >Streptococcus jiangjianxini </td>\n",
       "      <td id=\"T_2c4c2_row6_col1\" class=\"data row6 col1\" >Streptococcus jiangjianxini</td>\n",
       "      <td id=\"T_2c4c2_row6_col2\" class=\"data row6 col2\" >ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99</td>\n",
       "      <td id=\"T_2c4c2_row6_col3\" class=\"data row6 col3\" >JBEHGJ000000000</td>\n",
       "      <td id=\"T_2c4c2_row6_col4\" class=\"data row6 col4\" >ZJ151</td>\n",
       "      <td id=\"T_2c4c2_row6_col5\" class=\"data row6 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row6_col6\" class=\"data row6 col6\" >3161189.000000</td>\n",
       "      <td id=\"T_2c4c2_row6_col7\" class=\"data row6 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row6_col8\" class=\"data row6 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "      <td id=\"T_2c4c2_row7_col0\" class=\"data row7 col0\" >Streptococcus jiangjianxini </td>\n",
       "      <td id=\"T_2c4c2_row7_col1\" class=\"data row7 col1\" >Streptococcus jiangjianxini</td>\n",
       "      <td id=\"T_2c4c2_row7_col2\" class=\"data row7 col2\" >ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99</td>\n",
       "      <td id=\"T_2c4c2_row7_col3\" class=\"data row7 col3\" >JBEHGL000000000</td>\n",
       "      <td id=\"T_2c4c2_row7_col4\" class=\"data row7 col4\" >ZJ99</td>\n",
       "      <td id=\"T_2c4c2_row7_col5\" class=\"data row7 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row7_col6\" class=\"data row7 col6\" >3161189.000000</td>\n",
       "      <td id=\"T_2c4c2_row7_col7\" class=\"data row7 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row7_col8\" class=\"data row7 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "      <td id=\"T_2c4c2_row8_col0\" class=\"data row8 col0\" >Streptococcus handemini </td>\n",
       "      <td id=\"T_2c4c2_row8_col1\" class=\"data row8 col1\" >Streptococcus handemini</td>\n",
       "      <td id=\"T_2c4c2_row8_col2\" class=\"data row8 col2\" >ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93</td>\n",
       "      <td id=\"T_2c4c2_row8_col3\" class=\"data row8 col3\" >JBEHGM000000000</td>\n",
       "      <td id=\"T_2c4c2_row8_col4\" class=\"data row8 col4\" >ZJ100</td>\n",
       "      <td id=\"T_2c4c2_row8_col5\" class=\"data row8 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row8_col6\" class=\"data row8 col6\" >3161188.000000</td>\n",
       "      <td id=\"T_2c4c2_row8_col7\" class=\"data row8 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row8_col8\" class=\"data row8 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "      <td id=\"T_2c4c2_row9_col0\" class=\"data row9 col0\" >Aestuariibius violaceus </td>\n",
       "      <td id=\"T_2c4c2_row9_col1\" class=\"data row9 col1\" >Aestuariibius violaceus</td>\n",
       "      <td id=\"T_2c4c2_row9_col2\" class=\"data row9 col2\" >2305UL40-4T,  KCCM 43505T,  MCCC 1K09164T</td>\n",
       "      <td id=\"T_2c4c2_row9_col3\" class=\"data row9 col3\" >JBFONT000000000</td>\n",
       "      <td id=\"T_2c4c2_row9_col4\" class=\"data row9 col4\" >2305UL40-4</td>\n",
       "      <td id=\"T_2c4c2_row9_col5\" class=\"data row9 col5\" >Lee et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row9_col6\" class=\"data row9 col6\" >3234132.000000</td>\n",
       "      <td id=\"T_2c4c2_row9_col7\" class=\"data row9 col7\" >https://doi.org/10.1099/ijsem.0.006834</td>\n",
       "      <td id=\"T_2c4c2_row9_col8\" class=\"data row9 col8\" >http://dx.doi.org/10.1099/ijsem.0.006834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row10\" class=\"row_heading level0 row10\" >10</th>\n",
       "      <td id=\"T_2c4c2_row10_col0\" class=\"data row10 col0\" >Streptococcus huangxiaojuni </td>\n",
       "      <td id=\"T_2c4c2_row10_col1\" class=\"data row10 col1\" >Streptococcus huangxiaojuni</td>\n",
       "      <td id=\"T_2c4c2_row10_col2\" class=\"data row10 col2\" >H31T,  GDMCC 1.4758T,  JCM 37397T,  H49</td>\n",
       "      <td id=\"T_2c4c2_row10_col3\" class=\"data row10 col3\" >JBFQEG000000000</td>\n",
       "      <td id=\"T_2c4c2_row10_col4\" class=\"data row10 col4\" >H31</td>\n",
       "      <td id=\"T_2c4c2_row10_col5\" class=\"data row10 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row10_col6\" class=\"data row10 col6\" >3237239.000000</td>\n",
       "      <td id=\"T_2c4c2_row10_col7\" class=\"data row10 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row10_col8\" class=\"data row10 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row11\" class=\"row_heading level0 row11\" >11</th>\n",
       "      <td id=\"T_2c4c2_row11_col0\" class=\"data row11 col0\" >Streptococcus huangxiaojuni </td>\n",
       "      <td id=\"T_2c4c2_row11_col1\" class=\"data row11 col1\" >Streptococcus huangxiaojuni</td>\n",
       "      <td id=\"T_2c4c2_row11_col2\" class=\"data row11 col2\" >H31T,  GDMCC 1.4758T,  JCM 37397T,  H49</td>\n",
       "      <td id=\"T_2c4c2_row11_col3\" class=\"data row11 col3\" >JBFQEH000000000</td>\n",
       "      <td id=\"T_2c4c2_row11_col4\" class=\"data row11 col4\" >H49</td>\n",
       "      <td id=\"T_2c4c2_row11_col5\" class=\"data row11 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row11_col6\" class=\"data row11 col6\" >3237239.000000</td>\n",
       "      <td id=\"T_2c4c2_row11_col7\" class=\"data row11 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row11_col8\" class=\"data row11 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row12\" class=\"row_heading level0 row12\" >12</th>\n",
       "      <td id=\"T_2c4c2_row12_col0\" class=\"data row12 col0\" >Streptococcus handemini </td>\n",
       "      <td id=\"T_2c4c2_row12_col1\" class=\"data row12 col1\" >Streptococcus handemini</td>\n",
       "      <td id=\"T_2c4c2_row12_col2\" class=\"data row12 col2\" >ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93</td>\n",
       "      <td id=\"T_2c4c2_row12_col3\" class=\"data row12 col3\" >JBFRYH000000000</td>\n",
       "      <td id=\"T_2c4c2_row12_col4\" class=\"data row12 col4\" >ZJ93</td>\n",
       "      <td id=\"T_2c4c2_row12_col5\" class=\"data row12 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row12_col6\" class=\"data row12 col6\" >3161188.000000</td>\n",
       "      <td id=\"T_2c4c2_row12_col7\" class=\"data row12 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row12_col8\" class=\"data row12 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row13\" class=\"row_heading level0 row13\" >13</th>\n",
       "      <td id=\"T_2c4c2_row13_col0\" class=\"data row13 col0\" >Oceaniferula spumae </td>\n",
       "      <td id=\"T_2c4c2_row13_col1\" class=\"data row13 col1\" >Oceaniferula spumae</td>\n",
       "      <td id=\"T_2c4c2_row13_col2\" class=\"data row13 col2\" >NT6NT</td>\n",
       "      <td id=\"T_2c4c2_row13_col3\" class=\"data row13 col3\" >LC820896</td>\n",
       "      <td id=\"T_2c4c2_row13_col4\" class=\"data row13 col4\" >NT6N</td>\n",
       "      <td id=\"T_2c4c2_row13_col5\" class=\"data row13 col5\" >Huang et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row13_col6\" class=\"data row13 col6\" >2979115.000000</td>\n",
       "      <td id=\"T_2c4c2_row13_col7\" class=\"data row13 col7\" >https://doi.org/10.1099/ijsem.0.006828</td>\n",
       "      <td id=\"T_2c4c2_row13_col8\" class=\"data row13 col8\" >http://dx.doi.org/10.1099/ijsem.0.006828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row14\" class=\"row_heading level0 row14\" >14</th>\n",
       "      <td id=\"T_2c4c2_row14_col0\" class=\"data row14 col0\" >Fodinibius salipaludis </td>\n",
       "      <td id=\"T_2c4c2_row14_col1\" class=\"data row14 col1\" >Fodinibius salipaludis</td>\n",
       "      <td id=\"T_2c4c2_row14_col2\" class=\"data row14 col2\" >WN023T,  KCTC 52855T,  ACCC 19978T</td>\n",
       "      <td id=\"T_2c4c2_row14_col3\" class=\"data row14 col3\" >MF782427</td>\n",
       "      <td id=\"T_2c4c2_row14_col4\" class=\"data row14 col4\" >WN023</td>\n",
       "      <td id=\"T_2c4c2_row14_col5\" class=\"data row14 col5\" >He et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row14_col6\" class=\"data row14 col6\" >2032627.000000</td>\n",
       "      <td id=\"T_2c4c2_row14_col7\" class=\"data row14 col7\" >https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td id=\"T_2c4c2_row14_col8\" class=\"data row14 col8\" >http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row15\" class=\"row_heading level0 row15\" >15</th>\n",
       "      <td id=\"T_2c4c2_row15_col0\" class=\"data row15 col0\" >Streptococcus tangpeifui </td>\n",
       "      <td id=\"T_2c4c2_row15_col1\" class=\"data row15 col1\" >Streptococcus tangpeifui</td>\n",
       "      <td id=\"T_2c4c2_row15_col2\" class=\"data row15 col2\" >ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373</td>\n",
       "      <td id=\"T_2c4c2_row15_col3\" class=\"data row15 col3\" >MN850392</td>\n",
       "      <td id=\"T_2c4c2_row15_col4\" class=\"data row15 col4\" >ZJ373</td>\n",
       "      <td id=\"T_2c4c2_row15_col5\" class=\"data row15 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row15_col6\" class=\"data row15 col6\" >2709400.000000</td>\n",
       "      <td id=\"T_2c4c2_row15_col7\" class=\"data row15 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row15_col8\" class=\"data row15 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row16\" class=\"row_heading level0 row16\" >16</th>\n",
       "      <td id=\"T_2c4c2_row16_col0\" class=\"data row16 col0\" >Fodinibius salipaludis </td>\n",
       "      <td id=\"T_2c4c2_row16_col1\" class=\"data row16 col1\" >Fodinibius salipaludis</td>\n",
       "      <td id=\"T_2c4c2_row16_col2\" class=\"data row16 col2\" >WN023T,  KCTC 52855T,  ACCC 19978T</td>\n",
       "      <td id=\"T_2c4c2_row16_col3\" class=\"data row16 col3\" >NSKE00000000</td>\n",
       "      <td id=\"T_2c4c2_row16_col4\" class=\"data row16 col4\" >WN023</td>\n",
       "      <td id=\"T_2c4c2_row16_col5\" class=\"data row16 col5\" >He et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row16_col6\" class=\"data row16 col6\" >2032627.000000</td>\n",
       "      <td id=\"T_2c4c2_row16_col7\" class=\"data row16 col7\" >https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td id=\"T_2c4c2_row16_col8\" class=\"data row16 col8\" >http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row17\" class=\"row_heading level0 row17\" >17</th>\n",
       "      <td id=\"T_2c4c2_row17_col0\" class=\"data row17 col0\" >Streptococcus tangpeifui </td>\n",
       "      <td id=\"T_2c4c2_row17_col1\" class=\"data row17 col1\" >Streptococcus tangpeifui</td>\n",
       "      <td id=\"T_2c4c2_row17_col2\" class=\"data row17 col2\" >ZJ1593T,  GDMCC 1.3115T,  JCM 35327T,  ZJ373</td>\n",
       "      <td id=\"T_2c4c2_row17_col3\" class=\"data row17 col3\" >OL504562</td>\n",
       "      <td id=\"T_2c4c2_row17_col4\" class=\"data row17 col4\" >ZJ1593</td>\n",
       "      <td id=\"T_2c4c2_row17_col5\" class=\"data row17 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row17_col6\" class=\"data row17 col6\" >2709400.000000</td>\n",
       "      <td id=\"T_2c4c2_row17_col7\" class=\"data row17 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row17_col8\" class=\"data row17 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row18\" class=\"row_heading level0 row18\" >18</th>\n",
       "      <td id=\"T_2c4c2_row18_col0\" class=\"data row18 col0\" >Streptococcus jiangjianxini </td>\n",
       "      <td id=\"T_2c4c2_row18_col1\" class=\"data row18 col1\" >Streptococcus jiangjianxini</td>\n",
       "      <td id=\"T_2c4c2_row18_col2\" class=\"data row18 col2\" >ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99</td>\n",
       "      <td id=\"T_2c4c2_row18_col3\" class=\"data row18 col3\" >OL504968</td>\n",
       "      <td id=\"T_2c4c2_row18_col4\" class=\"data row18 col4\" >ZJ151</td>\n",
       "      <td id=\"T_2c4c2_row18_col5\" class=\"data row18 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row18_col6\" class=\"data row18 col6\" >3161189.000000</td>\n",
       "      <td id=\"T_2c4c2_row18_col7\" class=\"data row18 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row18_col8\" class=\"data row18 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row19\" class=\"row_heading level0 row19\" >19</th>\n",
       "      <td id=\"T_2c4c2_row19_col0\" class=\"data row19 col0\" >Streptococcus handemini </td>\n",
       "      <td id=\"T_2c4c2_row19_col1\" class=\"data row19 col1\" >Streptococcus handemini</td>\n",
       "      <td id=\"T_2c4c2_row19_col2\" class=\"data row19 col2\" >ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93</td>\n",
       "      <td id=\"T_2c4c2_row19_col3\" class=\"data row19 col3\" >OL504969</td>\n",
       "      <td id=\"T_2c4c2_row19_col4\" class=\"data row19 col4\" >ZJ93</td>\n",
       "      <td id=\"T_2c4c2_row19_col5\" class=\"data row19 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row19_col6\" class=\"data row19 col6\" >3161188.000000</td>\n",
       "      <td id=\"T_2c4c2_row19_col7\" class=\"data row19 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row19_col8\" class=\"data row19 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row20\" class=\"row_heading level0 row20\" >20</th>\n",
       "      <td id=\"T_2c4c2_row20_col0\" class=\"data row20 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row20_col1\" class=\"data row20 col1\" >Farysia sp.</td>\n",
       "      <td id=\"T_2c4c2_row20_col2\" class=\"data row20 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row20_col3\" class=\"data row20 col3\" >OP566881</td>\n",
       "      <td id=\"T_2c4c2_row20_col4\" class=\"data row20 col4\" >NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row20_col5\" class=\"data row20 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row20_col6\" class=\"data row20 col6\" >2012362.000000</td>\n",
       "      <td id=\"T_2c4c2_row20_col7\" class=\"data row20 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row20_col8\" class=\"data row20 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row21\" class=\"row_heading level0 row21\" >21</th>\n",
       "      <td id=\"T_2c4c2_row21_col0\" class=\"data row21 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row21_col1\" class=\"data row21 col1\" >Farysia sp.</td>\n",
       "      <td id=\"T_2c4c2_row21_col2\" class=\"data row21 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row21_col3\" class=\"data row21 col3\" >OP566885</td>\n",
       "      <td id=\"T_2c4c2_row21_col4\" class=\"data row21 col4\" >NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row21_col5\" class=\"data row21 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row21_col6\" class=\"data row21 col6\" >2012362.000000</td>\n",
       "      <td id=\"T_2c4c2_row21_col7\" class=\"data row21 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row21_col8\" class=\"data row21 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row22\" class=\"row_heading level0 row22\" >22</th>\n",
       "      <td id=\"T_2c4c2_row22_col0\" class=\"data row22 col0\" >Methanosuratincola petrocarbonis </td>\n",
       "      <td id=\"T_2c4c2_row22_col1\" class=\"data row22 col1\" >Methanosuratincola petrocarbonis</td>\n",
       "      <td id=\"T_2c4c2_row22_col2\" class=\"data row22 col2\" >LWZ-6T,  CCAM 1872T,  JCM 39528T</td>\n",
       "      <td id=\"T_2c4c2_row22_col3\" class=\"data row22 col3\" >OR243905</td>\n",
       "      <td id=\"T_2c4c2_row22_col4\" class=\"data row22 col4\" >LWZ-6</td>\n",
       "      <td id=\"T_2c4c2_row22_col5\" class=\"data row22 col5\" >Wu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row22_col6\" class=\"data row22 col6\" >3448697.000000</td>\n",
       "      <td id=\"T_2c4c2_row22_col7\" class=\"data row22 col7\" >https://doi.org/10.1099/ijsem.0.006839</td>\n",
       "      <td id=\"T_2c4c2_row22_col8\" class=\"data row22 col8\" >http://dx.doi.org/10.1099/ijsem.0.006839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row23\" class=\"row_heading level0 row23\" >23</th>\n",
       "      <td id=\"T_2c4c2_row23_col0\" class=\"data row23 col0\" >Aestuariibius violaceus </td>\n",
       "      <td id=\"T_2c4c2_row23_col1\" class=\"data row23 col1\" >Aestuariibius violaceus</td>\n",
       "      <td id=\"T_2c4c2_row23_col2\" class=\"data row23 col2\" >2305UL40-4T,  KCCM 43505T,  MCCC 1K09164T</td>\n",
       "      <td id=\"T_2c4c2_row23_col3\" class=\"data row23 col3\" >OR945532</td>\n",
       "      <td id=\"T_2c4c2_row23_col4\" class=\"data row23 col4\" >2305UL40-4</td>\n",
       "      <td id=\"T_2c4c2_row23_col5\" class=\"data row23 col5\" >Lee et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row23_col6\" class=\"data row23 col6\" >3234132.000000</td>\n",
       "      <td id=\"T_2c4c2_row23_col7\" class=\"data row23 col7\" >https://doi.org/10.1099/ijsem.0.006834</td>\n",
       "      <td id=\"T_2c4c2_row23_col8\" class=\"data row23 col8\" >http://dx.doi.org/10.1099/ijsem.0.006834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row24\" class=\"row_heading level0 row24\" >24</th>\n",
       "      <td id=\"T_2c4c2_row24_col0\" class=\"data row24 col0\" >Chlamydia vaughanii </td>\n",
       "      <td id=\"T_2c4c2_row24_col1\" class=\"data row24 col1\" >Chlamydia vaughanii</td>\n",
       "      <td id=\"T_2c4c2_row24_col2\" class=\"data row24 col2\" >ATCC CRL-1696,  EPC,  ATCC CRL-2872</td>\n",
       "      <td id=\"T_2c4c2_row24_col3\" class=\"data row24 col3\" >OZ026853</td>\n",
       "      <td id=\"T_2c4c2_row24_col4\" class=\"data row24 col4\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row24_col5\" class=\"data row24 col5\" >Marquis et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row24_col6\" class=\"data row24 col6\" >3112552.000000</td>\n",
       "      <td id=\"T_2c4c2_row24_col7\" class=\"data row24 col7\" >https://doi.org/10.6084/m9.figshare.28424588.v</td>\n",
       "      <td id=\"T_2c4c2_row24_col8\" class=\"data row24 col8\" >http://dx.doi.org/10.1099/ijsem.0.006753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row25\" class=\"row_heading level0 row25\" >25</th>\n",
       "      <td id=\"T_2c4c2_row25_col0\" class=\"data row25 col0\" >Brachymonas wangyanguii </td>\n",
       "      <td id=\"T_2c4c2_row25_col1\" class=\"data row25 col1\" >Brachymonas sp.</td>\n",
       "      <td id=\"T_2c4c2_row25_col2\" class=\"data row25 col2\" >G13T,  GDMCC 1.4691T,  JCM 37746T</td>\n",
       "      <td id=\"T_2c4c2_row25_col3\" class=\"data row25 col3\" >PP212021</td>\n",
       "      <td id=\"T_2c4c2_row25_col4\" class=\"data row25 col4\" >G13</td>\n",
       "      <td id=\"T_2c4c2_row25_col5\" class=\"data row25 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row25_col6\" class=\"data row25 col6\" >1936292.000000</td>\n",
       "      <td id=\"T_2c4c2_row25_col7\" class=\"data row25 col7\" >https://doi.org/10.1099/ijsem.0.006837</td>\n",
       "      <td id=\"T_2c4c2_row25_col8\" class=\"data row25 col8\" >http://dx.doi.org/10.1099/ijsem.0.006837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row26\" class=\"row_heading level0 row26\" >26</th>\n",
       "      <td id=\"T_2c4c2_row26_col0\" class=\"data row26 col0\" >Fodinibius alkaliphilus </td>\n",
       "      <td id=\"T_2c4c2_row26_col1\" class=\"data row26 col1\" >Fodinibius alkaliphilus</td>\n",
       "      <td id=\"T_2c4c2_row26_col2\" class=\"data row26 col2\" >N2T,  KCTC 102228T,  MCCC 1K08942T</td>\n",
       "      <td id=\"T_2c4c2_row26_col3\" class=\"data row26 col3\" >PP669793</td>\n",
       "      <td id=\"T_2c4c2_row26_col4\" class=\"data row26 col4\" >N2</td>\n",
       "      <td id=\"T_2c4c2_row26_col5\" class=\"data row26 col5\" >He et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row26_col6\" class=\"data row26 col6\" >3140241.000000</td>\n",
       "      <td id=\"T_2c4c2_row26_col7\" class=\"data row26 col7\" >https://doi.org/10.1099/ijsem.0.006840</td>\n",
       "      <td id=\"T_2c4c2_row26_col8\" class=\"data row26 col8\" >http://dx.doi.org/10.1099/ijsem.0.006840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row27\" class=\"row_heading level0 row27\" >27</th>\n",
       "      <td id=\"T_2c4c2_row27_col0\" class=\"data row27 col0\" >Streptococcus jiangjianxini </td>\n",
       "      <td id=\"T_2c4c2_row27_col1\" class=\"data row27 col1\" >Streptococcus jiangjianxini</td>\n",
       "      <td id=\"T_2c4c2_row27_col2\" class=\"data row27 col2\" >ZJ151T,  GDMCC 1.3116T,  JCM 35326T,  ZJ99</td>\n",
       "      <td id=\"T_2c4c2_row27_col3\" class=\"data row27 col3\" >PP830068</td>\n",
       "      <td id=\"T_2c4c2_row27_col4\" class=\"data row27 col4\" >ZJ99</td>\n",
       "      <td id=\"T_2c4c2_row27_col5\" class=\"data row27 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row27_col6\" class=\"data row27 col6\" >3161189.000000</td>\n",
       "      <td id=\"T_2c4c2_row27_col7\" class=\"data row27 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row27_col8\" class=\"data row27 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row28\" class=\"row_heading level0 row28\" >28</th>\n",
       "      <td id=\"T_2c4c2_row28_col0\" class=\"data row28 col0\" >Streptococcus handemini </td>\n",
       "      <td id=\"T_2c4c2_row28_col1\" class=\"data row28 col1\" >Streptococcus handemini</td>\n",
       "      <td id=\"T_2c4c2_row28_col2\" class=\"data row28 col2\" >ZJ93T,  GDMCC 1.3112T,  JCM 35325T,  ZJ93</td>\n",
       "      <td id=\"T_2c4c2_row28_col3\" class=\"data row28 col3\" >PP830069</td>\n",
       "      <td id=\"T_2c4c2_row28_col4\" class=\"data row28 col4\" >ZJ100</td>\n",
       "      <td id=\"T_2c4c2_row28_col5\" class=\"data row28 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row28_col6\" class=\"data row28 col6\" >3161188.000000</td>\n",
       "      <td id=\"T_2c4c2_row28_col7\" class=\"data row28 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row28_col8\" class=\"data row28 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row29\" class=\"row_heading level0 row29\" >29</th>\n",
       "      <td id=\"T_2c4c2_row29_col0\" class=\"data row29 col0\" >Streptococcus huangxiaojuni </td>\n",
       "      <td id=\"T_2c4c2_row29_col1\" class=\"data row29 col1\" >Streptococcus huangxiaojuni</td>\n",
       "      <td id=\"T_2c4c2_row29_col2\" class=\"data row29 col2\" >H31T,  GDMCC 1.4758T,  JCM 37397T,  H49</td>\n",
       "      <td id=\"T_2c4c2_row29_col3\" class=\"data row29 col3\" >PQ001808</td>\n",
       "      <td id=\"T_2c4c2_row29_col4\" class=\"data row29 col4\" >H31</td>\n",
       "      <td id=\"T_2c4c2_row29_col5\" class=\"data row29 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row29_col6\" class=\"data row29 col6\" >3237239.000000</td>\n",
       "      <td id=\"T_2c4c2_row29_col7\" class=\"data row29 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row29_col8\" class=\"data row29 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row30\" class=\"row_heading level0 row30\" >30</th>\n",
       "      <td id=\"T_2c4c2_row30_col0\" class=\"data row30 col0\" >Legionella sheltonii </td>\n",
       "      <td id=\"T_2c4c2_row30_col1\" class=\"data row30 col1\" >Legionella sheltonii</td>\n",
       "      <td id=\"T_2c4c2_row30_col2\" class=\"data row30 col2\" >11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td id=\"T_2c4c2_row30_col3\" class=\"data row30 col3\" >PQ119488</td>\n",
       "      <td id=\"T_2c4c2_row30_col4\" class=\"data row30 col4\" >PATHC038</td>\n",
       "      <td id=\"T_2c4c2_row30_col5\" class=\"data row30 col5\" >Klobucar et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row30_col6\" class=\"data row30 col6\" >2992041.000000</td>\n",
       "      <td id=\"T_2c4c2_row30_col7\" class=\"data row30 col7\" >https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td id=\"T_2c4c2_row30_col8\" class=\"data row30 col8\" >http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row31\" class=\"row_heading level0 row31\" >31</th>\n",
       "      <td id=\"T_2c4c2_row31_col0\" class=\"data row31 col0\" >Legionella sheltonii </td>\n",
       "      <td id=\"T_2c4c2_row31_col1\" class=\"data row31 col1\" >Legionella sheltonii</td>\n",
       "      <td id=\"T_2c4c2_row31_col2\" class=\"data row31 col2\" >11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td id=\"T_2c4c2_row31_col3\" class=\"data row31 col3\" >PQ119489</td>\n",
       "      <td id=\"T_2c4c2_row31_col4\" class=\"data row31 col4\" >HB10</td>\n",
       "      <td id=\"T_2c4c2_row31_col5\" class=\"data row31 col5\" >Klobucar et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row31_col6\" class=\"data row31 col6\" >2992041.000000</td>\n",
       "      <td id=\"T_2c4c2_row31_col7\" class=\"data row31 col7\" >https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td id=\"T_2c4c2_row31_col8\" class=\"data row31 col8\" >http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row32\" class=\"row_heading level0 row32\" >32</th>\n",
       "      <td id=\"T_2c4c2_row32_col0\" class=\"data row32 col0\" >Legionella sheltonii </td>\n",
       "      <td id=\"T_2c4c2_row32_col1\" class=\"data row32 col1\" >Legionella sheltonii</td>\n",
       "      <td id=\"T_2c4c2_row32_col2\" class=\"data row32 col2\" >11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td id=\"T_2c4c2_row32_col3\" class=\"data row32 col3\" >PQ120582</td>\n",
       "      <td id=\"T_2c4c2_row32_col4\" class=\"data row32 col4\" >HB10</td>\n",
       "      <td id=\"T_2c4c2_row32_col5\" class=\"data row32 col5\" >Klobucar et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row32_col6\" class=\"data row32 col6\" >2992041.000000</td>\n",
       "      <td id=\"T_2c4c2_row32_col7\" class=\"data row32 col7\" >https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td id=\"T_2c4c2_row32_col8\" class=\"data row32 col8\" >http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row33\" class=\"row_heading level0 row33\" >33</th>\n",
       "      <td id=\"T_2c4c2_row33_col0\" class=\"data row33 col0\" >Legionella sheltonii </td>\n",
       "      <td id=\"T_2c4c2_row33_col1\" class=\"data row33 col1\" >Legionella sheltonii</td>\n",
       "      <td id=\"T_2c4c2_row33_col2\" class=\"data row33 col2\" >11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td id=\"T_2c4c2_row33_col3\" class=\"data row33 col3\" >PQ120583</td>\n",
       "      <td id=\"T_2c4c2_row33_col4\" class=\"data row33 col4\" >PATHC038</td>\n",
       "      <td id=\"T_2c4c2_row33_col5\" class=\"data row33 col5\" >Klobucar et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row33_col6\" class=\"data row33 col6\" >2992041.000000</td>\n",
       "      <td id=\"T_2c4c2_row33_col7\" class=\"data row33 col7\" >https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td id=\"T_2c4c2_row33_col8\" class=\"data row33 col8\" >http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row34\" class=\"row_heading level0 row34\" >34</th>\n",
       "      <td id=\"T_2c4c2_row34_col0\" class=\"data row34 col0\" >Streptococcus huangxiaojuni </td>\n",
       "      <td id=\"T_2c4c2_row34_col1\" class=\"data row34 col1\" >Streptococcus huangxiaojuni</td>\n",
       "      <td id=\"T_2c4c2_row34_col2\" class=\"data row34 col2\" >H31T,  GDMCC 1.4758T,  JCM 37397T,  H49</td>\n",
       "      <td id=\"T_2c4c2_row34_col3\" class=\"data row34 col3\" >PQ198380</td>\n",
       "      <td id=\"T_2c4c2_row34_col4\" class=\"data row34 col4\" >H49</td>\n",
       "      <td id=\"T_2c4c2_row34_col5\" class=\"data row34 col5\" >Pu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row34_col6\" class=\"data row34 col6\" >3237239.000000</td>\n",
       "      <td id=\"T_2c4c2_row34_col7\" class=\"data row34 col7\" >https://doi.org/10.1099/ijsem.0.006814</td>\n",
       "      <td id=\"T_2c4c2_row34_col8\" class=\"data row34 col8\" >http://dx.doi.org/10.1099/ijsem.0.006814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row35\" class=\"row_heading level0 row35\" >35</th>\n",
       "      <td id=\"T_2c4c2_row35_col0\" class=\"data row35 col0\" >F.Y. Bai </td>\n",
       "      <td id=\"T_2c4c2_row35_col1\" class=\"data row35 col1\" >Rhodotorula sp.</td>\n",
       "      <td id=\"T_2c4c2_row35_col2\" class=\"data row35 col2\" >CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3</td>\n",
       "      <td id=\"T_2c4c2_row35_col3\" class=\"data row35 col3\" >PQ432547</td>\n",
       "      <td id=\"T_2c4c2_row35_col4\" class=\"data row35 col4\" >1017-17-3</td>\n",
       "      <td id=\"T_2c4c2_row35_col5\" class=\"data row35 col5\" >Wei et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row35_col6\" class=\"data row35 col6\" >1853554.000000</td>\n",
       "      <td id=\"T_2c4c2_row35_col7\" class=\"data row35 col7\" >https://doi.org/10.1099/ijsem.0.006836</td>\n",
       "      <td id=\"T_2c4c2_row35_col8\" class=\"data row35 col8\" >http://dx.doi.org/10.1099/ijsem.0.006836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row36\" class=\"row_heading level0 row36\" >36</th>\n",
       "      <td id=\"T_2c4c2_row36_col0\" class=\"data row36 col0\" >F.Y. Bai </td>\n",
       "      <td id=\"T_2c4c2_row36_col1\" class=\"data row36 col1\" >Rhodotorula sp.</td>\n",
       "      <td id=\"T_2c4c2_row36_col2\" class=\"data row36 col2\" >CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3</td>\n",
       "      <td id=\"T_2c4c2_row36_col3\" class=\"data row36 col3\" >PQ432548</td>\n",
       "      <td id=\"T_2c4c2_row36_col4\" class=\"data row36 col4\" >1017-17-3</td>\n",
       "      <td id=\"T_2c4c2_row36_col5\" class=\"data row36 col5\" >Wei et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row36_col6\" class=\"data row36 col6\" >1853554.000000</td>\n",
       "      <td id=\"T_2c4c2_row36_col7\" class=\"data row36 col7\" >https://doi.org/10.1099/ijsem.0.006836</td>\n",
       "      <td id=\"T_2c4c2_row36_col8\" class=\"data row36 col8\" >http://dx.doi.org/10.1099/ijsem.0.006836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row37\" class=\"row_heading level0 row37\" >37</th>\n",
       "      <td id=\"T_2c4c2_row37_col0\" class=\"data row37 col0\" >Brachymonas wangyanguii </td>\n",
       "      <td id=\"T_2c4c2_row37_col1\" class=\"data row37 col1\" >Brachymonas sp.</td>\n",
       "      <td id=\"T_2c4c2_row37_col2\" class=\"data row37 col2\" >G13T,  GDMCC 1.4691T,  JCM 37746T</td>\n",
       "      <td id=\"T_2c4c2_row37_col3\" class=\"data row37 col3\" >PQ632502</td>\n",
       "      <td id=\"T_2c4c2_row37_col4\" class=\"data row37 col4\" >G34</td>\n",
       "      <td id=\"T_2c4c2_row37_col5\" class=\"data row37 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row37_col6\" class=\"data row37 col6\" >1936292.000000</td>\n",
       "      <td id=\"T_2c4c2_row37_col7\" class=\"data row37 col7\" >https://doi.org/10.1099/ijsem.0.006837</td>\n",
       "      <td id=\"T_2c4c2_row37_col8\" class=\"data row37 col8\" >http://dx.doi.org/10.1099/ijsem.0.006837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row38\" class=\"row_heading level0 row38\" >38</th>\n",
       "      <td id=\"T_2c4c2_row38_col0\" class=\"data row38 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row38_col1\" class=\"data row38 col1\" >Farysizyma sp.</td>\n",
       "      <td id=\"T_2c4c2_row38_col2\" class=\"data row38 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row38_col3\" class=\"data row38 col3\" >PV170746</td>\n",
       "      <td id=\"T_2c4c2_row38_col4\" class=\"data row38 col4\" >NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row38_col5\" class=\"data row38 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row38_col6\" class=\"data row38 col6\" >2650697.000000</td>\n",
       "      <td id=\"T_2c4c2_row38_col7\" class=\"data row38 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row38_col8\" class=\"data row38 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row39\" class=\"row_heading level0 row39\" >39</th>\n",
       "      <td id=\"T_2c4c2_row39_col0\" class=\"data row39 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row39_col1\" class=\"data row39 col1\" >Farysizyma sp.</td>\n",
       "      <td id=\"T_2c4c2_row39_col2\" class=\"data row39 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row39_col3\" class=\"data row39 col3\" >PV170747</td>\n",
       "      <td id=\"T_2c4c2_row39_col4\" class=\"data row39 col4\" >NYNU 229246</td>\n",
       "      <td id=\"T_2c4c2_row39_col5\" class=\"data row39 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row39_col6\" class=\"data row39 col6\" >2650697.000000</td>\n",
       "      <td id=\"T_2c4c2_row39_col7\" class=\"data row39 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row39_col8\" class=\"data row39 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row40\" class=\"row_heading level0 row40\" >40</th>\n",
       "      <td id=\"T_2c4c2_row40_col0\" class=\"data row40 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row40_col1\" class=\"data row40 col1\" >Farysizyma sp.</td>\n",
       "      <td id=\"T_2c4c2_row40_col2\" class=\"data row40 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row40_col3\" class=\"data row40 col3\" >PV170748</td>\n",
       "      <td id=\"T_2c4c2_row40_col4\" class=\"data row40 col4\" >NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row40_col5\" class=\"data row40 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row40_col6\" class=\"data row40 col6\" >2650697.000000</td>\n",
       "      <td id=\"T_2c4c2_row40_col7\" class=\"data row40 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row40_col8\" class=\"data row40 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row41\" class=\"row_heading level0 row41\" >41</th>\n",
       "      <td id=\"T_2c4c2_row41_col0\" class=\"data row41 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row41_col1\" class=\"data row41 col1\" >Farysizyma sp.</td>\n",
       "      <td id=\"T_2c4c2_row41_col2\" class=\"data row41 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row41_col3\" class=\"data row41 col3\" >PV170749</td>\n",
       "      <td id=\"T_2c4c2_row41_col4\" class=\"data row41 col4\" >NYNU 229246</td>\n",
       "      <td id=\"T_2c4c2_row41_col5\" class=\"data row41 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row41_col6\" class=\"data row41 col6\" >2650697.000000</td>\n",
       "      <td id=\"T_2c4c2_row41_col7\" class=\"data row41 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row41_col8\" class=\"data row41 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row42\" class=\"row_heading level0 row42\" >42</th>\n",
       "      <td id=\"T_2c4c2_row42_col0\" class=\"data row42 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row42_col1\" class=\"data row42 col1\" >Farysizyma sp.</td>\n",
       "      <td id=\"T_2c4c2_row42_col2\" class=\"data row42 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row42_col3\" class=\"data row42 col3\" >PV170750</td>\n",
       "      <td id=\"T_2c4c2_row42_col4\" class=\"data row42 col4\" >NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row42_col5\" class=\"data row42 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row42_col6\" class=\"data row42 col6\" >2650697.000000</td>\n",
       "      <td id=\"T_2c4c2_row42_col7\" class=\"data row42 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row42_col8\" class=\"data row42 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row43\" class=\"row_heading level0 row43\" >43</th>\n",
       "      <td id=\"T_2c4c2_row43_col0\" class=\"data row43 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row43_col1\" class=\"data row43 col1\" >Farysizyma sp.</td>\n",
       "      <td id=\"T_2c4c2_row43_col2\" class=\"data row43 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row43_col3\" class=\"data row43 col3\" >PV170751</td>\n",
       "      <td id=\"T_2c4c2_row43_col4\" class=\"data row43 col4\" >NYNU 229246</td>\n",
       "      <td id=\"T_2c4c2_row43_col5\" class=\"data row43 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row43_col6\" class=\"data row43 col6\" >2650697.000000</td>\n",
       "      <td id=\"T_2c4c2_row43_col7\" class=\"data row43 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row43_col8\" class=\"data row43 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row44\" class=\"row_heading level0 row44\" >44</th>\n",
       "      <td id=\"T_2c4c2_row44_col0\" class=\"data row44 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row44_col1\" class=\"data row44 col1\" >Farysia sp.</td>\n",
       "      <td id=\"T_2c4c2_row44_col2\" class=\"data row44 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row44_col3\" class=\"data row44 col3\" >PV172327</td>\n",
       "      <td id=\"T_2c4c2_row44_col4\" class=\"data row44 col4\" >NYNU 229246</td>\n",
       "      <td id=\"T_2c4c2_row44_col5\" class=\"data row44 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row44_col6\" class=\"data row44 col6\" >2012362.000000</td>\n",
       "      <td id=\"T_2c4c2_row44_col7\" class=\"data row44 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row44_col8\" class=\"data row44 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row45\" class=\"row_heading level0 row45\" >45</th>\n",
       "      <td id=\"T_2c4c2_row45_col0\" class=\"data row45 col0\" >Farysia photiniae </td>\n",
       "      <td id=\"T_2c4c2_row45_col1\" class=\"data row45 col1\" >Farysia sp.</td>\n",
       "      <td id=\"T_2c4c2_row45_col2\" class=\"data row45 col2\" >PYCC 9932,  NYNU 22992</td>\n",
       "      <td id=\"T_2c4c2_row45_col3\" class=\"data row45 col3\" >PV172328</td>\n",
       "      <td id=\"T_2c4c2_row45_col4\" class=\"data row45 col4\" >NYNU 229246</td>\n",
       "      <td id=\"T_2c4c2_row45_col5\" class=\"data row45 col5\" >Gao et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row45_col6\" class=\"data row45 col6\" >2012362.000000</td>\n",
       "      <td id=\"T_2c4c2_row45_col7\" class=\"data row45 col7\" >https://doi.org/10.1099/ijsem.0.006850</td>\n",
       "      <td id=\"T_2c4c2_row45_col8\" class=\"data row45 col8\" >http://dx.doi.org/10.1099/ijsem.0.006850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row46\" class=\"row_heading level0 row46\" >46</th>\n",
       "      <td id=\"T_2c4c2_row46_col0\" class=\"data row46 col0\" >F.Y. Bai </td>\n",
       "      <td id=\"T_2c4c2_row46_col1\" class=\"data row46 col1\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row46_col2\" class=\"data row46 col2\" >CGMCC 2.7770,  UCDFST 81-84,  PDA,  YM,  CGMCC 2.7770T,  1017-17-3</td>\n",
       "      <td id=\"T_2c4c2_row46_col3\" class=\"data row46 col3\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row46_col4\" class=\"data row46 col4\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row46_col5\" class=\"data row46 col5\" >Wei et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row46_col6\" class=\"data row46 col6\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row46_col7\" class=\"data row46 col7\" >https://doi.org/10.1099/ijsem.0.006836</td>\n",
       "      <td id=\"T_2c4c2_row46_col8\" class=\"data row46 col8\" >http://dx.doi.org/10.1099/ijsem.0.006836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row47\" class=\"row_heading level0 row47\" >47</th>\n",
       "      <td id=\"T_2c4c2_row47_col0\" class=\"data row47 col0\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row47_col1\" class=\"data row47 col1\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row47_col2\" class=\"data row47 col2\" ></td>\n",
       "      <td id=\"T_2c4c2_row47_col3\" class=\"data row47 col3\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row47_col4\" class=\"data row47 col4\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row47_col5\" class=\"data row47 col5\" >Wu et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row47_col6\" class=\"data row47 col6\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row47_col7\" class=\"data row47 col7\" >https://doi.org/10.1099/ijsem.0.006839</td>\n",
       "      <td id=\"T_2c4c2_row47_col8\" class=\"data row47 col8\" >http://dx.doi.org/10.1099/ijsem.0.006839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_2c4c2_level0_row48\" class=\"row_heading level0 row48\" >48</th>\n",
       "      <td id=\"T_2c4c2_row48_col0\" class=\"data row48 col0\" >Legionella sheltonii </td>\n",
       "      <td id=\"T_2c4c2_row48_col1\" class=\"data row48 col1\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row48_col2\" class=\"data row48 col2\" >11976T,  PATHC038T,  HB10,  CCUG 76918T,  ATCC TSD-370</td>\n",
       "      <td id=\"T_2c4c2_row48_col3\" class=\"data row48 col3\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row48_col4\" class=\"data row48 col4\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row48_col5\" class=\"data row48 col5\" >Klobucar et al. 2025</td>\n",
       "      <td id=\"T_2c4c2_row48_col6\" class=\"data row48 col6\" >nan</td>\n",
       "      <td id=\"T_2c4c2_row48_col7\" class=\"data row48 col7\" >https://doi.org/10.1099/ijsem.0.006813</td>\n",
       "      <td id=\"T_2c4c2_row48_col8\" class=\"data row48 col8\" >http://dx.doi.org/10.1099/ijsem.0.006813</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x7f708b345a50>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def highlight_rows(row):\n",
    "    ijsemvalue = row.loc['PublishedName']\n",
    "    ncbivalue = row.loc['NCBIname']\n",
    "    if ijsemvalue != ncbivalue:\n",
    "        color = '#FFB3BA' # Red\n",
    "    elif ijsemvalue == ncbivalue:\n",
    "        color = '#BAFFC9' # Green\n",
    "    return ['background-color: {}'.format(color) for r in row]\n",
    "\n",
    "new_df = combine_df.style.apply(highlight_rows, axis=1, subset=['PublishedName', 'NCBIname'])\n",
    "new_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5aa05366",
   "metadata": {},
   "source": [
    "### write output to excel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "fb8948c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_df.to_excel(output, engine='xlsxwriter', index = False, na_rep = '') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ca925cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "combine_df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "652c0d09",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b75b8410",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d865873",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "da2c3095",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3e319d0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12912624",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f19f3af8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7380ffa1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7591cab8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88ffd0f6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08a7f93d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
